---
title: "Clustering annotation of E10.5 Integrated data to regress sex-specific genes"
author: "Ekpereka Amutaigwe"
date: "2023-05-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load packages

```{r}
suppressPackageStartupMessages ({
library(here)
library(SingleCellExperiment)
library(scater)
library(Seurat)
library(ggplot2)
library(Matrix)
library(reticulate)
#library(ggpubr) # For donut plot
#library(cowplot)
library(EnhancedVolcano)
library(textshaping)
library(RCurl)
library(umap)
library(R.utils)
library(ensembldb)
library(AnnotationHub)
#library(dyno)
library(tidyverse)
library(pheatmap)
#library(biomaRt)
#library(EnsDb.Mmusculus.v79)
library(dittoSeq)
library(ComplexHeatmap)
#library(scCustomize)
library(patchwork)
library(gridExtra)
library(grid)
library(scales)
library(RColorBrewer)
#library(ggsci)# Package for scientific journal color palettes
#library(pals) # for colours
#library(monocle3)
#library(SeuratWrappers)
#library(slingshot)
#library(tradeSeq)
#library(scry)# recommended by Slingshot author for feature selection for running fitGAM on a subset of genes only instead of highly #variable genes from Seurat
#library(gam)
library(openxlsx)
})
```


```{r}
#dir.create("~/myproject/sc_analysis_project/msc_project/results/plots/FL_E12.5/12May_plots")
```


```{r, eval=FALSE}
# Read in Fetal liver E10.5 data
#fl_samples_E10.5_12May_notintegrated <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_notintegrated.rds")# Raw object. CHECK FOR BATCH EFFECT
fl_samples_E10.5_12May <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/E10.5_integrated.rds") # Raw object.
fl_samples_E10.5_12May <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_clustered_sexgenesnotremoved.rds")
#fl_samples_E10.5_12May_named <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named.rds")
fl_samples_E10.5_12May_named <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_MERGED_EPs_Mks.rds")
markers_fl_samples_E10.5_12May_named <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/markers_fl_samples_E10.5_12May_named.rds")
fl_samples_E10.5_12May_named_subset <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset.rds")
fl_samples_E10.5_12May_named_subset_DEG <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset_DEG.rds")
```


# The cells do not cluster based on sex. Do not regress out sex genes 
```{r}
#fl_samples_E10.5_12May_regressed <- fl_samples_E10.5_12May
#rm(fl_samples_E10.5_12May)

#DefaultAssay(fl_samples_E10.5_12May_regressed) <- "RNA"
```



```{r}
#DefaultAssay(fl_samples_E10.5_12May_regressed) <- "RNA"
#fl_samples_E10.5_12May_regressed <- AddModuleScore(object = fl_samples_E10.5_12May_regressed, features = sex_genes, name = "Sex_Genes")
#fl_samples_E10.5_12May_regressed[]


```


```{r}
# SCTransform the subset Ery object
#DefaultAssay(fl_samples_E10.5_12May_regressed) <- "RNA"
#fl_samples_E10.5_12May_regressed <- SCTransform(fl_samples_E10.5_12May_regressed,
                                   #vst.flavor = "v2",
                                   #vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score", "prediction"),
                                   #variable.features.n = 3000,
                                   #verbose = FALSE)
```



## Visualize subsets

```{r, eval=FALSE}
set.seed(1263)
DefaultAssay(fl_samples_E10.5_12May) <- "integrated"
# Run PCA
fl_samples_E10.5_12May<- RunPCA(object = fl_samples_E10.5_12May, verbose = FALSE)
```



```{r}
# Plot PCA
PCAPlot(fl_samples_E10.5_12May)

```


```{r, eval=FALSE}
# Plot the elbow plot
ElbowPlot(object = fl_samples_E10.5_12May, 
          ndims = 50)

```


```{r, eval=FALSE}
set.seed(122345)
# Run UMAP
fl_samples_E10.5_12May <- RunUMAP(fl_samples_E10.5_12May, 
                  dims = 1:50,
                  reduction = "pca", verbose = FALSE)

# Determine the K-nearest neighbour graph
fl_samples_E10.5_12May <- FindNeighbors(object = fl_samples_E10.5_12May,
                                  dims = 1:50, verbose = FALSE)

# Determine the clusters of the various resolutions
fl_samples_E10.5_12May  <- FindClusters(object = fl_samples_E10.5_12May,
                         resolution = c(0.5, 0.6, 0.7, 0.8, 0.9),
                                  algorithm = 1, verbose = FALSE)

# Assign identity to clusters
Idents(object = fl_samples_E10.5_12May) <- "integrated_snn_res.0.9"
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```


### Find subclusters of cluster 10

```{r, eval=FALSE}
# Find subclusters
fl_samples_E10.5_12May <- FindSubCluster(
  fl_samples_E10.5_12May,
  cluster = 10,
  graph.name = "integrated_snn",
  subcluster.name = "subcluster10",
  resolution = 0.2,
  algorithm = 1
) # Algorithm = 4 is Leiden algorithm

DimPlot(fl_samples_E10.5_12May, reduction = "umap", group.by = "subcluster10", label = TRUE, label.size = 6)
```



```{r, eval=FALSE}
# Update the cluster ident column in the seurat object
fl_samples_E10.5_12May <- SetIdent(fl_samples_E10.5_12May, value = fl_samples_E10.5_12May@meta.data$subcluster10)
```



### Find subclusters of cluster 11

```{r, eval=FALSE}
# Find subclusters
fl_samples_E10.5_12May<- FindSubCluster(
  fl_samples_E10.5_12May,
  cluster = 11,
  graph.name = "integrated_snn",
  subcluster.name = "subcluster11",
  resolution = 0.2,
  algorithm = 1
) # Algorithm = 4 is Leiden algorithm

DimPlot(fl_samples_E10.5_12May, reduction = "umap", group.by = "subcluster11", label = TRUE, label.size = 6)
```


```{r, eval=FALSE}
# Update the cluster ident column in the seurat object
fl_samples_E10.5_12May <- SetIdent(fl_samples_E10.5_12May, value = fl_samples_E10.5_12May@meta.data$subcluster11)
```


### Find subclusters of cluster 8

```{r, eval=FALSE}
# Find subclusters
fl_samples_E10.5_12May<- FindSubCluster(
  fl_samples_E10.5_12May,
  cluster = 8,
  graph.name = "integrated_snn",
  subcluster.name = "subcluster8",
  resolution = 0.2,
  algorithm = 1
) # Algorithm = 4 is Leiden algorithm

DimPlot(fl_samples_E10.5_12May, reduction = "umap", group.by = "subcluster8", label = TRUE, label.size = 6)
```


```{r, eval=FALSE}
# Update the cluster ident column in the seurat object
fl_samples_E10.5_12May <- SetIdent(fl_samples_E10.5_12May, value = fl_samples_E10.5_12May@meta.data$subcluster8)
```



### Find subclusters of cluster 7

```{r, eval=FALSE}
# Find subclusters
fl_samples_E10.5_12May<- FindSubCluster(
  fl_samples_E10.5_12May,
  cluster = 7,
  graph.name = "integrated_snn",
  subcluster.name = "subcluster7",
  resolution = 0.2,
  algorithm = 1
) # Algorithm = 4 is Leiden algorithm

DimPlot(fl_samples_E10.5_12May, reduction = "umap", group.by = "subcluster7", label = TRUE, label.size = 6)
```


```{r, eval=FALSE}
# Update the cluster ident column in the seurat object
fl_samples_E10.5_12May <- SetIdent(fl_samples_E10.5_12May, value = fl_samples_E10.5_12May@meta.data$subcluster7)
```


### Find subclusters of cluster 12

```{r, eval=FALSE}
# Find subclusters
fl_samples_E10.5_12May<- FindSubCluster(
  fl_samples_E10.5_12May,
  cluster = 12,
  graph.name = "integrated_snn",
  subcluster.name = "subcluster12",
  resolution = 0.2,
  algorithm = 1
) # Algorithm = 4 is Leiden algorithm

DimPlot(fl_samples_E10.5_12May, reduction = "umap", group.by = "subcluster12", label = TRUE, label.size = 6)
```


```{r, eval=FALSE}
# Update the cluster ident column in the seurat object
fl_samples_E10.5_12May <- SetIdent(fl_samples_E10.5_12May, value = fl_samples_E10.5_12May@meta.data$subcluster12)
```




```{r}
# Plot UMAP
set.seed(223)
DimPlot(object = fl_samples_E10.5_12May,
        reduction = "umap",
        label = TRUE,
        group.by = "sample",
        label.size = 6,
        pt.size = 1.5)

ggsave("repeat_grouped_unnamed_fl_samples_E10.5_12May.pdf", 
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```





```{r}
# Run PrepSCTFindMarkers to ensure that the fixed value is set properly, as SCT creates corrected counts by setting the sequencing depth for all the cells to a fixed value
DefaultAssay(fl_samples_E10.5_12May) <- "integrated"
fl_samples_E10.5_12May <- PrepSCTFindMarkers(fl_samples_E10.5_12May) 
```



## Identification of all markers for each cluster(This takes a while to run. Therefore, only use this option if you have just one condition)

```{r, eval=FALSE}
DefaultAssay(fl_samples_E10.5_12May) <- "RNA"
# code from the internet on how to remove specific genes before downstream analysis (Satijalab, Github)
counts <- GetAssayData(fl_samples_E10.5_12May, assay = "SCT")
counts <- counts[-(which(rownames(counts) %in% c("mt-Atp6", "mt-Nd1", "mt-Co1", "mt-Cytb",  "mt-Co2",  "mt-Nd4", "mt-Co3", "mt-Nd2", "mt-Nd4l", "mt-Atp8", "mt-Nd5", "mt-Nd3", "mt-Cytb1", "mt-Co31", "mt-Nd21",  "mt-Nd51",  "mt-Nd41",  "mt-Nd11", "mt-Nd4l1", "mt-Atp81", "mt-Co32"))),]
fl_samples_E10.5_12May <- subset(fl_samples_E10.5_12May, features = rownames(counts))
```


```{r, eval=FALSE}
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers_fl_samples_E10.5_12May <- FindAllMarkers(object = fl_samples_E10.5_12May, 
                          only.pos = TRUE,
                          assay = "SCT",
                          logfc.threshold = 0.25,
                          min.pct = 0.25,
                          thresh.use = 0.25)
```



```{r, eval=FALSE} 
# Arrange markers by average log2fold change
markers_fl_samples_E10.5_12May <- markers_fl_samples_E10.5_12May %>% 
  dplyr::group_by(cluster) %>% 
  dplyr::arrange(desc(avg_log2FC), .by_group = TRUE)

```


```{r}
# Let us look at the top markers of each cluster

top_3_fl_samples_E10.5_12May <- markers_fl_samples_E10.5_12May %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 3)

#top_30_markers_fl_samples_E10.5_12May <- markers_fl_samples_E10.5_12May %>% group_by(cluster) %>% 
  #slice_max(avg_log2FC, n = 30)
```


```{r}
DefaultAssay(fl_samples_E10.5_12May) <- "RNA"
```


```{r}
# Megakaryocyte progenitors
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("F2r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) 
  
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Itga2b"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Gp1bb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
a
b
c

```

```{r}
# Megakaryocytes
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Myl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Vwf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Ppbp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
d <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Alox12"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
e <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Zyx"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('MEGAKARYOCYTE')))
a
b
c
d
e
```


```{r}
# Mono/Mac
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Ccr2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Csf1r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b,c),ncol = 3, nrow=1, top= ('MONOCYTE/KUPFFER')))
a
c
```

```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Gata2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Klf1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Megakaryocyte-Erythroid progenitors')))
a
b
c
```


```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Epor"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Car1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Muc13"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Megakaryocyte-Erythroid progenitors')))
a
b
c
```




```{r}
# Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroid progenitors Kit pos, Gata1 pos')))
a
b
```


```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Hbb-bs"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Hbb-bt"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Gypa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroblasts - Tfrc high, Kit low/neg, Bpgm low/neg')))
a
b
c
```


```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroblasts - Tfrc high, Kit low/neg, Bpgm low/neg')))
a
b
c
```


```{r}
# Hepatoblasts
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Afp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Alb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Hepatoblasts Afp pos, Alb pos')))
a
b
```


```{r}
# Endothelial cells
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Kdr"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Cldn5"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Endothelial cells Cldn5 pos, Cdh5 pos')))
a
b
```


```{r}
# Mesenchymal cells
a <- FeaturePlot(fl_samples_E10.5_12May, 
            reduction = "umap", 
            features = c("Alcam"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
a

```






```{r}
# Plot top 3 markers per object 
DotPlot(
  fl_samples_E10.5_12May,
  assay = NULL,
  features = unique(top_3_fl_samples_E10.5_12May$gene),
  cols = c("lightgrey", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 10, face = "bold")) + ggtitle("FL_E10.5 Top 3 marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")



known_markers1 <- c("Gypa", "Tfrc", "Gata1", "Kit", "Bpgm", "Hba-x", "Hbb-y", "Ptn", "Gata2", "Afp", "Alb", "Hnf4a", "Cdh5", "Cldn5", "Spi1", "Csf1r", "Pf4", "Vwf", "Elane", "Hp", "Cx3cr1", "Ifitm1", "Irf8", "Itga2b", "Prtn3", "Thbs1", "Gata2", "Gata1", "Klf1", "Car1", "Cd34", "Fcgr3", "Ccr2", "C1qa", "C1qb", "Prg2", "Stx3", "Nnat", "Pdgfra", "Il7r", "Cd7", "Hlf", "Mecom", "Ly6a", "Prg2", "Ppbp", "Ngp", "Hdc", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Ccr2", "Hdc", "Cpa3", "Gp1bb", "Cd24a", "Gp9", "Muc13", "Hlf", "Hoxa9", "Hoxa7")
  

  
  
known_markers1  <- markers_fl_samples_E10.5_12May %>% 
                dplyr::filter(gene %in% known_markers1)

DotPlot(
  fl_samples_E10.5_12May,
  assay = NULL,
  features = unique(known_markers1$gene),
  cols = c("darkblue", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 10, face = "bold")) + ggtitle("FL_E10.5 - Known marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")


```


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```

```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May,
        reduction = "umap",
        label = FALSE,
        group.by = "Phase",
        label.size = 6,
        pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size =50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
            theme(legend.text = element_text(size = 50, face = "bold")) +
            guides(color = guide_legend(override.aes = list(size = 5))) +
            theme(axis.text = element_text(size = 50, face = "bold"), 
            panel.background = element_rect(fill = "white"), 
            plot.background = element_rect(fill = "white", colour = "black", linewidth = 1)) 
ggsave("repeated_phase_umap2_fl_samples_E10.5_12May.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May,
        reduction = "umap",
        label = FALSE,
        group.by = "Phase",
        label.size = 6,
        pt.size = 1)
```



## Annotate clusters in FL_E10.5 samples
```{r}
# Fetal liver
fl_samples_E10.5_12May_named <-  RenameIdents(object = fl_samples_E10.5_12May,
                                      "0" = "EP1",
                                      "1" = "MEP",
                                      "2" = "EP3",
                                      "3" = "Mk1",
                                      "4" = "EP2",
                                      "5" = "Prim-EB",
                                      "6" = "MEP",
                                      "7_0" = "PEB1",
                                      "7_1" = "PEB2",
                                      "7_2" = "PEB3",
                                      "8_0" = "HSC/MPP",
                                      "8_1" = "HSC/MPP",
                                      "9" = "Mk2",
                                      "10_0" = "Mono",
                                      "10_1" = "Mac",
                                      "11_0" = "MCP",
                                      "11_1" = "GMP/GP",
                                      "12_0" = "Mesench",
                                      "12_1" = "Endo",
                                      "13" = "Hepato")

fl_samples_E10.5_12May_named$cluster_ident <- Idents(fl_samples_E10.5_12May_named)

```




```{r}
# Determine total number of hematopoietic cells
table(fl_samples_E10.5_12May_named$cluster_ident, fl_samples_E10.5_12May_named$sample_group)
table(fl_samples_E10.5_12May_named$sample_group, fl_samples_E10.5_12May_named$sample_group)
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```


```{r}
#DefaultAssay(fl_samples_E10.5_12May_named) <- "SCT"
```


## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E10.5_12May_named <- subset(fl_samples_E10.5_12May_named, idents =c("HSC/MPP", "GMP/GP", "Mono", "Mac", "MCP", "MEP", "Mk1", "Mk2", "Prim-EB","EP1", "EP2", "EP3","PEB1", "PEB2", "PEB3"))
```



```{r}
DefaultAssay(fl_samples_E10.5_12May_named) <- "RNA"
# SCTransform the subset Ery object
fl_samples_E10.5_12May_named <- SCTransform(fl_samples_E10.5_12May_named,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



## Reannotate clusters in FL_E10.5 samples
```{r}
# Fetal liver
fl_samples_E10.5_12May_named <-  RenameIdents(object = fl_samples_E10.5_12May_named,
                                      "EP1" = "EP",
                                      "EP3" = "EP",
                                      "Mk1" = "Mk",
                                      "EP2" = "EP",
                                      "PEB1" = "PEB",
                                      "PEB2" = "PEB",
                                      "PEB3" = "PEB",
                                      "HSC/MPP" = "HSC/EMP",
                                      "Mk2" = "Mk"
)

fl_samples_E10.5_12May_named$cluster_ident <- Idents(fl_samples_E10.5_12May_named)

```





```{r}
fl_samples_E10.5_12May_named@active.ident <- factor(fl_samples_E10.5_12May_named@active.ident, levels = c("HSC/EMP", "GMP/GP", "Mono", "Mac", "MCP", "Prim-EB", "MEP", "Mk", "EP", "PEB"))
```


```{r, eval=FALSE}
fl_samples_E10.5_12May_named$cluster_ident <- factor(fl_samples_E10.5_12May_named$cluster_ident, levels = c("HSC/EMP", "GMP/GP", "Mono", "Mac", "MCP", "Prim-EB", "MEP", "Mk", "EP", "PEB"))
```




```{r}
# Run PrepSCTFindMarkers to ensure that the fixed value is set properly, as SCT creates corrected counts by setting the sequencing depth for all the cells to a fixed value
DefaultAssay(fl_samples_E10.5_12May_named) <- "integrated"
fl_samples_E10.5_12May_named <- PrepSCTFindMarkers(fl_samples_E10.5_12May_named) 
```


```{r, eval=FALSE}
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers_fl_samples_E10.5_12May_named <- FindAllMarkers(object = fl_samples_E10.5_12May_named, 
                          only.pos = TRUE,
                          assay = "SCT",
                          logfc.threshold = 0.25,
                          min.pct = 0.25,
                          thresh.use = 0.25)
```


```{r, eval=FALSE} 
# Arrange markers by average log2fold change
markers_fl_samples_E10.5_12May_named <- markers_fl_samples_E10.5_12May_named %>% 
  dplyr::group_by(cluster) %>% 
  dplyr::arrange(desc(avg_log2FC), .by_group = TRUE)


write.csv(markers_fl_samples_E10.5_12May_named, file = paste0(here("results/FL_E10.5/"), "markers_fl_samples_E10.5_12May_named.csv"))
```


```{r}
# Let us look at the top markers of each cluster

top_3_markers_fl_samples_E10.5_12May_named <- markers_fl_samples_E10.5_12May_named %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 3)
top_5_markers_fl_samples_E10.5_12May_named <- markers_fl_samples_E10.5_12May_named %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 5)
top_10_markers_fl_samples_E10.5_12May_named <- markers_fl_samples_E10.5_12May_named %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 10)
```


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```




```{r}
# Set colors
clust_cols <- c(PEB = "#16FF32",  Mk = "#F8A19F", MCP = "#DEA0FD", `Prim-EB` = "#1C8356", EP = "#90AD1C", Mono = "#FBE426", Mac = "darkorange2", `GMP/GP` = "grey60", MEP = "#F6222E", `HSC/EMP` = "#00AFBB")
```



```{r}
PEry2 = "#90AD1C", "dodgerblue3", Hepato = "#3283FE", Mesench = , Endo = ,EMP1.MPP1= "#B00068",MkP = "#00AFBB",
EP1 = "#16FF32", EP2 = "#F7E1A0",Mk1 = , "#E2E2E2"PEB1 = "#2ED9FF", PEB3 = "#FE00FA", "#1CFFCE""#3283FE"
```

 

```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        label = TRUE,
        cols = clust_cols,
        label.size = 6,
        pt.size = 1.5)
```


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        cols = clust_cols,
        label = FALSE,
        label.size = 11,
        pt.size = 1.5) + 
        #NoLegend() +
        #theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.ticks = element_blank()) +
        theme(axis.line = element_blank()) +
        #panel.background = element_rect(fill = "white") +
        theme(plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) 
ggsave("repeated_umap2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
(ncells_E10.5 <- data.frame(table(fl_samples_E10.5_12May_named$cluster_ident)) %>% dplyr::rename("Clusters" = Var1))

(ncells_E10.5 <- mutate(ncells_E10.5, percent = round((Freq/sum(Freq))*100, digits = 2)))
ncells_E10.5$percent <- paste0(ncells_E10.5$percent,"%")
ncells_E10.5

ggplot(ncells_E10.5, aes(x = Clusters, y = Freq, fill = Clusters)) +
  geom_col() +
  scale_fill_manual(values = c(PEB = "#16FF32",  Mk = "#F8A19F", MCP = "#DEA0FD", EP = "#90AD1C", Mono = "#FBE426", Mac = "darkorange2", `GMP/GP` = "grey60", MEP = "#F6222E",  `Prim-EB` = "#1C8356", `HSC/EMP` = "#00AFBB")) +
  theme(axis.text = element_text(size = 50),
        panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) +
        theme(axis.title.x = element_text(size =50, face = "bold")) +
        theme(axis.title.y = element_text(size = 50, face = "bold")) +
  geom_text(aes(label = percent), vjust = 0.5, hjust = -0.1, size = 15, face = "bold") +
  #scale_y_continuous(breaks = seq(0, 2000, by=1000), limits=c(0,2000)) +
  scale_y_continuous(breaks = seq(0, 4000, by=1000), limits=c(0,4000)) +
  xlab("Cell types") + 
 # theme_bw() + 
  NoLegend() +
  coord_flip()

ggsave("repeated_cellsincluster_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
DefaultAssay(fl_samples_E10.5_12May_named) <- "RNA"

# First, subset object into replicates
E10.5_rep1 <- subset(x = fl_samples_E10.5_12May_named, subset = sample == "FL_E10.5_rep_1")
E10.5_rep2 <- subset(x = fl_samples_E10.5_12May_named, subset = sample == "FL_E10.5_rep_2")

# Average genes per cell in each object
(ave_genes_percell_e10.5_rep1 <- mean(Matrix::colSums(GetAssayData(E10.5_rep1, slot = 'counts') > 0)))
(ave_genes_percell_e10.5_rep2 <- mean(Matrix::colSums(GetAssayData(E10.5_rep2, slot = 'counts') > 0)))

# Get average reads/counts per cell
(ave_counts_percell_e10.5_rep1 <- mean(E10.5_rep1@meta.data$nUMI))
(ave_counts_percell_e10.5_rep2 <- mean(E10.5_rep2@meta.data$nUMI))

rm(E10.5_rep1, E10.5_rep2)
```


[1] 4179.761
[1] 4294.002
[1] 19243.11
[1] 17704.9
```{r}
DefaultAssay(fl_samples_E10.5_12May_named) <- "RNA"
# HSC/EMP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hlf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) + 
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank()) 

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hoxa7"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hoxa9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ly6a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 2) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank()) 

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Mecom"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 2) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank()) 

f <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Fgd5"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 2) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

g <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Spi1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 2) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("EMP7_hlf_markers1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("EMP7_hoxa7_markers2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("EMP7_hoxa9_markers3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("EMP7_Ly6a_markers4_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")

e
ggsave("EMP7_Mecom_markers5_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
f
ggsave("EMP8_Fgd5_markers6_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
g
ggsave("EMP8_Spi1_markers7_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# EMP/MPP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Fcgr3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ccl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cd34"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cd93"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("EMP_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("EMP_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("EMP_marker3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("EMP_marker4_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("EMP_marker5_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# MPP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Prtn3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ms4a3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Mpo"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ptprc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("EMPalso_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("EMPalso_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("EMPalso_marker3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("EMPalso_marker4_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```

```{r}
# GMP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ms4a3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Mpo"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Perp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("GMP_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("GMP_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("GMP_marker3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("GMP_marker4_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}


# Megakaryocyte progenitors
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("F2r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Itga2b"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Gp1bb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
 
a
ggsave("R_Mk_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Mk_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_Mk_marker3_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Megakaryocytes
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Myl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Vwf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ppbp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Alox12"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Zyx"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("R_Mk_marker4_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Mk_marker5_fl_samples_E10.5_12May_amed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_Mk_marker6_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("R_Mk_marker8_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("R_Mk_marker7_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Gata2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Klf1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("R_MEP_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_MEP_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_MEP_marker3_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Epor"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Car1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Muc13"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
 
a
ggsave("R_MEP_marker4_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_MEP_marker5_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_MEP_marker6_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Erythroid progenitors
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Myb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("R_EryP_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_EryP_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_EryP_marker3_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("R_EryP_marker4_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("R_EryP_marker5_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hbb-bs"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hbb-bt"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Gypa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hba-a1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

f <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("R_Erythb_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Erythb_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("R_Erythb_marker3_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("R_Erythb_marker4_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("R_Erythb_marker5_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
f
ggsave("R_Erythb_marker6_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```


```{r}
# LP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Klrd1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Il7r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Rag1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ccr9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Flt3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("LP_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("LP_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("LP_marker3_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("LP_marker4_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("LP_marker5_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")

```


```{r}
# Myeloid
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ccl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Csf1r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Irf8"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("Myeloid_Ccl9_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Myeloid_Csf1r_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width =17, height = 15, device = "tiff")

c
ggsave("Myeloid_Irf8_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width =17, height = 15, device = "tiff")

```



```{r}
# Mono
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ccr2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cx3cr1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("Mono_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Mono_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width =17, height = 15, device = "tiff")

```




```{r}
# Mac
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("C1qa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Aif1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Mertk"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Adgre1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("Mac_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Mac_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("Mac_marker3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("Mac_marker4_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")

```

```{r}
# GP
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Elane"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Prg2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Prg3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("Gran_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Gran_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("Gran_marker3_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Mast cells
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cpa3"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hdc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("Mast_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Mast_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")

```

```{r}
# Primitive Erythrocytes
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hba-x"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Hbb-y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
 
a
ggsave("PEry_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15,  device = "tiff")
b
ggsave("PEry_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15,  device = "tiff")
```



```{r}
# Hepatoblasts
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Afp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Alb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_Hepato_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Hepato_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Endothelial cells
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cldn5"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Cdh5"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

a
ggsave("Endo_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Endo_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```

```{r}
# Mesenchymal cells
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ptn"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Alcam"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

a
ggsave("Mesench_marker1_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("Mesench_marker2_fl_samples_E10.5_12May_named.tiff",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




# LOOK AT SOME Y AND X CHROMOSOME GENES

```{r}
DefaultAssay(fl_samples_E10.5_12May_named) <- "SCT"
# Male genes
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Eif2s3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Ddx3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("R_Male_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Male_marker2_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Female genes
a <- FeaturePlot(fl_samples_E10.5_12May_named, 
            reduction = "umap", 
            features = c("Xist"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("R_Female_marker1_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
# Plot top 3 markers per object 
a <- DotPlot(
  fl_samples_E10.5_12May_named,
  assay = NULL,
  features = unique(top_3_markers_fl_samples_E10.5_12May_named$gene),
  cols = c("lightgrey", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 30, face = "bold")) + 
              theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
  theme(legend.text = element_text(size = 20, face = "bold")) +
  ggtitle("FL_E10.5 Top 3 marker genes") +
  theme(plot.title = element_text(face = "bold", size = 30)) + 
  theme(legend.title = element_text(size = 20), legend.position = "top") +
  scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")




known_markers1 <- c("Gypa", "Tfrc", "Gata1", "Kit", "Hba-x", "Hbb-y", "Gata2", "Spi1", "Csf1r", "Pf4", "Vwf", "Elane", "Hp", "Cx3cr1", "Ifitm1", "Irf8", "Itga2b", "Prtn3", "Thbs1", "Gata2", "Gata1", "Klf1", "Car1", "Cd34", "Fcgr3", "Ccr2", "C1qa", "C1qb", "Prg2", "Stx3", "Nnat", "Pdgfra", "Il7r", "Cd7", "Hlf", "Mecom", "Ly6a", "Prg2", "Ppbp", "Ngp", "Hdc", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Ccr2", "Hdc", "Cpa3", "Gp1bb", "Cd24a", "Ly6a", "Hoxa9", "Hoxa7", "Myb", "Epor", "Mpo", "Rhd", "Ms4a3", "Perp", "Muc13", "Plek", "F2r", "Cd93")
  

  
  
known_markers1  <- markers_fl_samples_E10.5_12May_named %>% 
                dplyr::filter(gene %in% known_markers1)

b <- DotPlot(
  fl_samples_E10.5_12May_named,
  assay = NULL,
  features = unique(known_markers1),
  cols = c("darkblue", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 30, face = "bold")) + 
              theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
  theme(legend.text = element_text(size = 20, face = "bold")) +
  theme(legend.title = element_text(size = 20), legend.position = "top") +
  ggtitle("FL_E10.5 - Known marker genes") +
  theme(plot.title = element_text(face = "bold", size = 30)) + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")


a
ggsave("Dotplot_marker1_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 45, height = 15,
       device = "pdf",
       dpi = 300
       )
b
ggsave("Dotplot_marker2_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 45, height = 15,
      device = "pdf",
       dpi = 300 )
```



```{r}
known_markers1 <- c("Gypa", "Tfrc", "Gata1", "Kit", "Hba-x", "Hbb-y", "Gata2", "Spi1", "Csf1r", "Pf4", "Vwf", "Elane", "Hp", "Cx3cr1", "Ifitm1", "Irf8", "Itga2b", "Prtn3", "Thbs1","Klf1", "Car1", "Cd34", "Fcgr3", "Ccr2", "C1qa", "C1qb", "Prg2", "Stx3", "Il7r", "Cd7", "Hlf", "Mecom","Prg2", "Ppbp", "Ngp", "Hdc", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Hdc", "Cpa3", "Gp1bb", "Cd24a","Hoxa9", "Hoxa7", "Myb", "Epor", "Mpo", "Rhd", "Ms4a3", "Perp", "Muc13", "Plek", "F2r", "Cd93")
```



```{r}
library(scCustomize)

#tiff("~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots/Clustered_Dotplot_marker2_fl_samples_E10.5_12May_named.tif", width = 600, height = 800)
Clustered_DotPlot(seurat_object = fl_samples_E10.5_12May_named, features = c("Gypa", "Tfrc", "Gata1", "Kit", "Hba-x", "Hbb-y", "Gata2", "Spi1", "Csf1r", "Pf4", "Vwf", "Elane", "Hp", "Cx3cr1", "Ifitm1", "Irf8", "Itga2b", "Prtn3", "Thbs1","Klf1", "Car1", "Cd34", "Fcgr3", "Ccr2", "C1qa", "C1qb", "Prg2", "Stx3", "Il7r", "Prg3", "Hlf", "Fgd5", "Mecom","Prg2", "Ppbp", "S100a8", "Hdc", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Hdc", "Cpa3", "Gp1bb", "Cd24a","Hoxa9", "Hoxa7", "Myb", "Epor", "Mpo", "Rhd", "Ms4a3", "Perp", "Muc13", "Plek", "Cd93", "Rag1", "Ccl9", "Ccr9", "Ly6a"), k = 9, show_parent_dend_line = FALSE, plot_km_elbow = FALSE, cluster_ident = FALSE, column_label_size = 10, row_label_size = 19, colors_use_idents = c(`HSC/EMP` = "#00AFBB", `GMP/GP` = "grey60", Mono = "#FBE426", Mac = "darkorange2", MCP = "#DEA0FD", `Prim-EB` = "#1C8356", MEP = "#F6222E", Mk = "#F8A19F",  EP = "#90AD1C", PEB = "#16FF32"), colors_use_exp = c(low="lightgrey", mid="grey", high="red"))

#dev.off()
```




```{r}
# Plot UMAP
set.seed(1203)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        cols = clust_cols,
        split.by = "sample",
        label.size = 6,
        pt.size = 1.5) +
        theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())
ggsave("R_replicates_color_umap_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E10.5_12May_named,
        reduction = "umap",
        label = FALSE,
        group.by = "sample",
        label.size = 6,
        pt.size = 1.5) +
        theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())
ggsave("R_by_sample_umap_of_replicates_fl_samples_E10.5_12May_amed.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```


```{r}
DefaultAssay(fl_samples_E10.5_12May_named) <- "RNA"
```


# Plot top 3 genes
```{r}
top_3_markers <- top_3_markers_fl_samples_E10.5_12May_named %>% 
                  ungroup() %>%
                  dplyr::select(gene) %>% 
                  unique() %>% 
                  .[!is.na(.)]

mat <- fl_samples_E10.5_12May_named@assays$SCT@data[top_3_markers, ] %>% as.matrix()
mat <- t(scale(t(mat)))

cols_sample_group <- c(FL_E10.5 = "#E2E2E2" )

clust <- c("HSC/EMP", "GMP/GP", "Mono", "Mac", "MCP", "Prim-EB", "MEP", "Mk", "EP", "PEB")

clusters <- fl_samples_E10.5_12May_named$cluster_ident

                
annotation_col <-  data.frame(E10.5=clusters)

annotation_colors <- list(E10.5 = clust_cols,
                          Timepoint=cols_sample_group)
                                        
# Determine expression ranges in the matrix
quantile(mat, c(0.1, 0.95))
cols <- circlize::colorRamp2(c(-1, 0, 2), c("darkgreen", "black", "yellow"))


#pdf(file="~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots/R_top3_heatmap_fl_samples_E10.5_12May_named.pdf", width = 25, height = 25)
heat <- ComplexHeatmap::Heatmap(mat, name="E10.5 Top3",
                        column_split = factor(clusters, levels = clust),
                        cluster_columns = TRUE,
                        show_column_dend = FALSE,
                        show_column_names = FALSE,
                        column_title = clust,
                        cluster_column_slices = FALSE,
                        column_title_gp = gpar(fontsize = 20),
                        row_names_gp = gpar(fontsize = 20),
                        column_gap = unit(0.5, "mm"),
                        cluster_rows = TRUE,
                        show_row_dend = FALSE,
                        col = cols,
                        row_gap = unit(10, "mm"),
                        column_title_rot = 60,
                        top_annotation = HeatmapAnnotation(df = annotation_col, col = annotation_colors, show_legend = FALSE, show_annotation_name = FALSE),
                        heatmap_legend_param = list(labels_gp = gpar(fontsize = 30), title_gp = gpar(fontsize = 30, fontface = "bold")),
                        width = 25,
                        height = 25,
                        use_raster = TRUE,
                        raster_quality = 4)
draw(heat)
#dev.off()
#annotation_legend_param = list(labels_gp = gpar(fontsize = 30),show_annotation_name = TRUE,title_gp = gpar(fontsize = 30, fontface = "bold")

```


```{r}
#library(viridis)
DefaultAssay(fl_samples_E10.5_12May_named) <- "RNA"
fl_samples_E10.5_12May_named <- ScaleData(fl_samples_E10.5_12May_named)
```




```{r}
#library(viridis)
DefaultAssay(fl_samples_E10.5_12May_named) <- "SCT"
mapal <- colorRampPalette(RColorBrewer::brewer.pal(11,"RdBu"))(256)
DoHeatmap(subset(fl_samples_E10.5_12May_named, downsample = 100),
          features = top_3_markers,
          group.by = "cluster_ident",
          group.colors = c(`HSC/EMP` = "#00AFBB", `GMP/GP` = "grey60", Mono = "#FBE426", Mac = "darkorange2", MCP = "#DEA0FD", `Prim-EB` = "#1C8356", MEP = "#F6222E", Mk = "#F8A19F",  EP = "#90AD1C", PEB = "#16FF32"),
          angle = 45) +
          theme(text = element_text(size = 30)) +
  scale_fill_gradientn(colours = rev(mapal))
          #scale_fill_viridis()
          
  ggsave("Top3_heatmap_fl_samples_E10.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




#Code copied from [https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/]
```{r}
## remove the x-axis text and tick
## plot.margin to adjust the white space between each plot.
## ... pass any arguments to VlnPlot in Seurat
modify_vlnplot<- function(obj, 
                          feature, 
                          pt.size = 0, 
                          plot.margin = unit(c(-0.75, 0, -0.75, 0), "cm"),
                          ...) {
  p<- VlnPlot(obj, features = feature, pt.size = pt.size, ... )  + 
    xlab("") + ylab(feature) + ggtitle("") + 
    theme(legend.position = "none", 
          axis.text.x = element_blank(),
          plot.title = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank(), 
          axis.title.y = element_text(size = rel(1), angle = 0, face = "bold"), 
          axis.text.y = element_text(size = rel(1)), 
          plot.margin = plot.margin ) + geom_boxplot(outlier.shape = NA) 
  return(p)
}

## extract the max value of the y axis
extract_max<- function(p){
  ymax<- max(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
  return(ceiling(ymax))
}


## main function
StackedVlnPlot<- function(obj, features,
                          pt.size = 0, 
                          plot.margin = unit(c(-0.75, 0, -0.75, 0), "cm"),
                          ...) {
  
  plot_list<- purrr::map(features, function(x) modify_vlnplot(obj = obj,feature = x, ...))
  
  # Add back x-axis title to bottom plot. patchwork is going to support this?
  plot_list[[length(plot_list)]]<- plot_list[[length(plot_list)]] +
    theme(axis.text.x=element_text(angle = 45, face = "bold"), axis.ticks.x = element_line())
  
  # change the y-axis tick to only max value 
  ymaxs<- purrr::map_dbl(plot_list, extract_max)
  plot_list<- purrr::map2(plot_list, ymaxs, function(x,y) x + 
                            scale_y_continuous(breaks = c(y)) + 
                            expand_limits(y = y))

  p<- patchwork::wrap_plots(plotlist = plot_list, ncol = 1)
  return(p)
}
```



```{r}
# MEP
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Gata1", "Car1", "Gata2", "Klf1", "Muc13", "Epor"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("MEP_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


```{r}
# Megakaryocytes
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Pf4", "Itga2b", "Ppbp", "Thbs1", "Gp9", "Zyx"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("Mk_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```
Zyx has the highest expression in megakaryocytes. It could be a new marker gene for identifying megakaryocytes.


```{r}
# Erythroid cells
StackedVlnPlot(obj = fl_samples_E14.5_12May_regressed_named, features = c("Myb", "Kit", "Gata1", "Tfrc", "Klf1", "Gypa", "Cd24a", "Epor", "Hba-a1", "Hba-a2", "Hbb-bs", "Hbb-bt", "Hbb-bh1"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("All_Ery_violin_fl_samples_E14.5_12May_regressed_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

# Look at Cd24a and Kit expression levels in the erythroid clusters
```{r}
# Erythroid cells
StackedVlnPlot(obj = fl_samples_E10.5_12May_named, features = c("Kit", "Tfrc", "Klf1", "Cd24a"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("Cd24a_Kit_violin_fl_samples_E10.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```





```{r}
# Hepatoblasts
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Afp", "Alb", "Ttr", "Hnf4a"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Hepato_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

```{r}
# Endothelial
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Cldn5", "Cdh5"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Endo_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

```{r}
# Mesenchymal cells
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Alcam", "Col1a2"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Mesench_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


```{r}
fl_samples_E10.5_12May_named_no_prim_ER <- subset(fl_samples_E10.5_12May_named, idents = c("HSC/EMP","GMP/GP", "Mono", "Mac", "MCP", "MEP", "Mk", "EP", "PEB"))
```



```{r}
fl_samples_E10.5_12May_named_no_prim_ER@active.ident <- factor(fl_samples_E10.5_12May_named_no_prim_ER@active.ident, levels = c("HSC/EMP","GMP/GP", "Mono", "Mac", "MCP", "MEP", "Mk", "EP","PEB"))
```


```{r, eval=FALSE}
fl_samples_E10.5_12May_named_no_prim_ER$cluster_ident <- factor(fl_samples_E10.5_12May_named_no_prim_ER$cluster_ident, levels = c("HSC/EMP","GMP/GP", "Mono", "Mac", "MCP", "MEP", "Mk", "EP","PEB"))
```



```{r}
saveRDS(fl_samples_E10.5_12May_named_no_prim_ER, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_no_prim_ER.rds")
```





## Subset Erythroid progenitors and regress out sex genes. THIS PART WAS DONE BEFORE ERYTHROID CLUSTERS WERE MERGED. DID NOT REPEAT THIS SINCE THE SAME SET OF ERYTHROID CLUSTERED WAS USED IN THIS ANALYSIS

## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E10.5_12May_named_subset_DEG <- subset(fl_samples_E10.5_12May_named_subset, idents =c("EP1", "EP2", "EP3"))
 
```



```{r}
# SCTransform the subset Ery object
fl_samples_E10.5_12May_named_subset_DEG <- SCTransform(fl_samples_E10.5_12May_named_subset_DEG,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



## Reannotate clusters in FL_E10.5 samples for DEG analysis between erythroid progenitors

```{r}
# Fetal liver
fl_samples_E10.5_12May_named_subset_DEG_merge <-  RenameIdents(object = fl_samples_E10.5_12May_named_subset_DEG,
                                                  "EP1" = "EP",
                                                  "EP2" = "EP",
                                                  "EP3" = "EP")

fl_samples_E10.5_12May_named_subset_DEG_merge$cluster_ident <- Idents(fl_samples_E10.5_12May_named_subset_DEG_merge)

```


```{r}
table(fl_samples_E10.5_12May_named_subset_DEG_merge$cluster_ident, fl_samples_E10.5_12May_named_subset_DEG_merge$sample_group)
```


```{r}
# Set colors
clust_cols_EP <- c(EP = "#16FF32")
```





```{r}
# Plot UMAP
set.seed(12098)
DimPlot(object = fl_samples_E10.5_12May_named_subset_DEG_merge,
        reduction = "umap",
        cols = clust_cols_EP,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend()

ggsave("R_umap_fl_samples_E10.5_12May_named_subset_DEG_merge.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E10.5/12May_plots",
       width = 17, height = 15)
```


```{r}
# Subset Cd45 (Ptprc) negative EPs
fl_samples_E10.5_12May_named_subset_DEG_merge_noCd45 <- subset(fl_samples_E10.5_12May_named_subset_DEG_merge, subset = Ptprc == 0)
```





```{r, eval=FALSE}
# Save integrated clustered and identified FL_E10.5 data

# Sept 11
saveRDS(fl_samples_E10.5_12May, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_clustered_sexgenesnotremoved.rds")
saveRDS(fl_samples_E10.5_12May_named, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named.rds") # Dec. 23
saveRDS(fl_samples_E10.5_12May_named, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_MERGED_EPs_Mks.rds") # Feb 25,2024
saveRDS(markers_fl_samples_E10.5_12May_named, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/markers_fl_samples_E10.5_12May_named.rds")
saveRDS(fl_samples_E10.5_12May_named_subset, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset.rds")
saveRDS(fl_samples_E10.5_12May_named_subset_DEG, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset_DEG.rds")
saveRDS(fl_samples_E10.5_12May_named_subset_DEG_merge, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset_DEG_merge.rds")
saveRDS(fl_samples_E10.5_12May_named_subset_DEG_merge_noCd45, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E10.5/fl_samples_E10.5_12May_named_subset_DEG_merge_noCd45.rds")
```



```{r}
sessionInfo()
```







