---
title: "Clustering annotation of E14.5 Integrated data to remove batch effect"
author: "Ekpereka Amutaigwe"
date: "2023-05-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load packages

```{r}
suppressPackageStartupMessages ({
library(here)
library(SingleCellExperiment)
library(scater)
library(Seurat)
library(ggplot2)
#library(Matrix)
library(reticulate)
#library(ggpubr) # For donut plot
library(cowplot)
library(EnhancedVolcano)
library(textshaping)
library(RCurl)
library(umap)
library(R.utils)
library(ensembldb)
library(AnnotationHub)
#library(dyno)
library(tidyverse)
library(pheatmap)
library(biomaRt)
library(EnsDb.Mmusculus.v79)
library(org.Mm.eg.db)
#library(dittoSeq)
library(ComplexHeatmap)
#library(scCustomize)
library(patchwork)
library(gridExtra)
library(grid)
library(scales)
library(RColorBrewer)
#library(ggsci)# Package for scientific journal color palettes
#library(pals) # for colours
#library(monocle3)
#library(SeuratWrappers)
#library(slingshot)
#library(tradeSeq)
#library(scry)# recommended by Slingshot author for feature selection for running fitGAM on a subset of genes only instead of highly #variable genes from Seurat
#library(gam)
library(openxlsx)
library(clusterProfiler)
library(msigdbr)
library(GOplot)
library(enrichplot)
library(AUCell)
})
```


```{r}
dir.create("~/myproject/sc_analysis_project/msc_project/results/plots/FL_E12.5/12May_plots")
```


```{r, eval=FALSE}
# Read in Fetal liver E14.5 data
#fl_samples_E14.5_12May_notintegrated <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_notintegrated.rds")# Raw object. CHECK FOR BATCH EFFECT
fl_samples_E14.5_12May <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/E14.5_integrated.rds") # Raw object.
fl_samples_E14.5_12May <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_clustered_sexgenesnotremoved.rds")
fl_samples_E14.5_12May_regressed <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_regressed.rds")
fl_samples_E14.5_12May_named_subset <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/repeated_fl_samples_E14.5_12May_named_subset.rds")
fl_samples_E14.5_12May_named_subset_regressed <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressedsexandsexgenes.rds")
markers_fl_samples_E14.5_12May_named_subset_regressed <- readRDS( "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/markers_fl_samples_E14.5_12May_named_subset_regressed.rds")
fl_samples_E14.5_12May_named_subset_regressed_named_EPs <- readRDS("~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressed_named_EPs.rds")
markers_fl_samples_E14.5_12May_named <- readRDS("~/myproject/sc_analysis_project/msc_project/results/FL_E14.5/markers_fl_samples_E14.5_12May_named.rds")
```



```{r}
#fl_samples_E14.5_12May_regressed <- fl_samples_E14.5_12May
#rm(fl_samples_E14.5_12May)

#DefaultAssay(fl_samples_E14.5_12May_regressed) <- "RNA"
```



## Visualize subsets

```{r, eval=FALSE}
set.seed(1263)
DefaultAssay(fl_samples_E14.5_12May) <- "integrated"
# Run PCA
fl_samples_E14.5_12May <- RunPCA(object = fl_samples_E14.5_12May, verbose = FALSE)
```



```{r}
# Plot PCA
PCAPlot(fl_samples_E14.5_12May)

```


```{r, eval=FALSE}
# Plot the elbow plot
ElbowPlot(object = fl_samples_E14.5_12May, 
          ndims = 50)

```


```{r, eval=FALSE}
set.seed(122345)
# Run UMAP
fl_samples_E14.5_12May <- RunUMAP(fl_samples_E14.5_12May, 
                  dims = 1:30,
                  reduction = "pca", verbose = FALSE)

# Determine the K-nearest neighbour graph
fl_samples_E14.5_12May <- FindNeighbors(object = fl_samples_E14.5_12May,
                                  dims = 1:30, verbose = FALSE)

# Determine the clusters of the various resolutions
fl_samples_E14.5_12May  <- FindClusters(object = fl_samples_E14.5_12May,
                         resolution = c(0.5, 0.6, 0.7, 0.8, 0.9),
                                  algorithm = 1, verbose = FALSE)

# Assign identity to clusters
Idents(object = fl_samples_E14.5_12May) <- "integrated_snn_res.0.7"
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)


```


```{r}
# Plot UMAP
set.seed(223)
DimPlot(object = fl_samples_E14.5_12May,
        reduction = "umap",
        label = TRUE,
        group.by = "sample",
        label.size = 6,
        pt.size = 1.5)

ggsave("repeat_grouped_unnamed_fl_samples_E14.5_12May.pdf", 
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```





```{r}
# Run PrepSCTFindMarkers to ensure that the fixed value is set properly, as SCT creates corrected counts by setting the sequencing depth for all the cells to a fixed value
DefaultAssay(fl_samples_E14.5_12May) <- "integrated"
fl_samples_E14.5_12May <- PrepSCTFindMarkers(fl_samples_E14.5_12May) 
```



## Identification of all markers for each cluster(This takes a while to run. Therefore, only use this option if you have just one condition)

```{r, eval=FALSE}
DefaultAssay(fl_samples_E14.5_12May) <- "SCT"
# code from the internet on how to remove specific genes before downstream analysis (Satijalab, Github)
counts <- GetAssayData(fl_samples_E14.5_12May, assay = "SCT")
counts <- counts[-(which(rownames(counts) %in% c("mt-Atp6", "mt-Nd1", "mt-Co1", "mt-Cytb",  "mt-Co2",  "mt-Nd4", "mt-Co3", "mt-Nd2", "mt-Nd4l", "mt-Atp8", "mt-Nd5", "mt-Nd3", "mt-Cytb1", "mt-Co31", "mt-Nd21",  "mt-Nd51",  "mt-Nd41",  "mt-Nd11", "mt-Nd4l1", "mt-Atp81", "mt-Co32"))),]
fl_samples_E14.5_12May <- subset(fl_samples_E14.5_12May, features = rownames(counts))
```


```{r, eval=FALSE}
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers_fl_samples_E14.5_12May <- FindAllMarkers(object = fl_samples_E14.5_12May, 
                          only.pos = TRUE,
                          assay = "SCT",
                          logfc.threshold = 0.25,
                          min.pct = 0.25,
                          thresh.use = 0.25)
```



```{r, eval=FALSE} 
# Arrange markers by average log2fold change
markers_fl_samples_E14.5_12May <- markers_fl_samples_E14.5_12May %>% 
  dplyr::group_by(cluster) %>% 
  dplyr::arrange(desc(avg_log2FC), .by_group = TRUE)

```


```{r}
# Let us look at the top markers of each cluster

top_3_fl_samples_E14.5_12May <- markers_fl_samples_E14.5_12May %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 3)

#top_30_markers_fl_samples_E14.5_12May <- markers_fl_samples_E14.5_12May %>% group_by(cluster) %>% 
  #slice_max(avg_log2FC, n = 30)
```


```{r}
DefaultAssay(fl_samples_E14.5_12May) <- "SCT"
```


```{r}
# Megakaryocyte progenitors
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("F2r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) 
  
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Itga2b"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Gp1bb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
a
b
c

```

```{r}
# Megakaryocytes
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Myl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Vwf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Ppbp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
d <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Alox12"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
e <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Zyx"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('MEGAKARYOCYTE')))
a
b
c
d
e
```


```{r}
# Mono/Mac
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Ccr2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Csf1r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b,c),ncol = 3, nrow=1, top= ('MONOCYTE/KUPFFER')))
a
c
```

```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Gata2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Klf1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Megakaryocyte-Erythroid progenitors')))
a
b
c
```


```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Epor"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Car1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Muc13"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
 
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Megakaryocyte-Erythroid progenitors')))
a
b
c
```




```{r}
# Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroid progenitors Kit pos, Gata1 pos')))
a
b
```


```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Hbb-bs"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Hbb-bt"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Gypa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroblasts - Tfrc high, Kit low/neg, Bpgm low/neg')))
a
b
c
```


```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Bpgm"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
c <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7)
#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Erythroblasts - Tfrc high, Kit low/neg, Bpgm low/neg')))
a
b
c
```


```{r}
# Hepatoblasts
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Afp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Alb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Hepatoblasts Afp pos, Alb pos')))
a
b
```


```{r}
# Endothelial cells
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Kdr"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
b <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Cldn5"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 

#print(gridExtra::marrangeGrob(list(a,b),ncol = 2, nrow=1, top= ('Endothelial cells Cldn5 pos, Cdh5 pos')))
a
b
```


```{r}
# Mesenchymal cells
a <- FeaturePlot(fl_samples_E14.5_12May, 
            reduction = "umap", 
            features = c("Alcam"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 0.7) 
a

```






```{r}
# Plot top 3 markers per object 
DotPlot(
  fl_samples_E14.5_12May,
  assay = NULL,
  features = unique(top_3_fl_samples_E14.5_12May$gene),
  cols = c("lightgrey", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 10, face = "bold")) + ggtitle("FL_E14.5 Top 3 marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")



known_markers1 <- c("Gypa", "Tfrc", "Gata1", "Kit", "Hba-x", "Hbb-y", "Ptn", "Gata2", "Afp", "Alb", "Hnf4a", "Cdh5", "Cldn5", "Spi1", "Csf1r", "Pf4", "Vwf", "Elane", "Hp", "Cx3cr1", "Ifitm1", "Irf8", "Itga2b", "Prtn3", "Thbs1", "Gata2", "Gata1", "Klf1", "Car1", "Cd34", "Fcgr3", "Ccr2", "C1qa", "C1qb", "Prg2", "Stx3", "Nnat", "Pdgfra", "Il7r", "Cd7", "Hlf", "Mecom", "Ly6a", "Prg2", "Ppbp", "Ngp", "Hdc", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Ccr2", "Hdc", "Cpa3", "Gp1bb", "Cd24a", "Gp9", "Muc13", "Hlf", "Hoxa9", "Hoxa7", "Nxpe2", "F2r", "Myl9", "Alox12", "Muc13")
  

  
  
known_markers1  <- markers_fl_samples_E14.5_12May %>% 
                dplyr::filter(gene %in% known_markers1)

DotPlot(
  fl_samples_E14.5_12May,
  assay = NULL,
  features = unique(known_markers1$gene),
  cols = c("darkblue", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + theme(axis.text = element_text(size = 10, face = "bold")) + ggtitle("FL_E12.5 - Known marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")


```

```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)


```



## Annotate clusters in FL_E12.5 samples
```{r}
# Fetal liver
fl_samples_E14.5_12May_named <-  RenameIdents(object = fl_samples_E14.5_12May,
                                      "0" = "EP1b",
                                      "1" = "EP2a",
                                      "2" = "EP3a",
                                      "3" = "PEBa",
                                      "4" = "EP2b",
                                      "5" = "PEBb",
                                      "6" = "EP1a",
                                      "7" = "PEBc",
                                      "8" = "EP4a",
                                      "9" = "MEP",
                                      "10" = "EP5a",
                                      "11" = "Mixed",
                                      "12" = "Mk",
                                      "13" = "Hepato")

fl_samples_E14.5_12May_named$cluster_ident <- Idents(fl_samples_E14.5_12May_named)


```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```


```{r}
DefaultAssay(fl_samples_E14.5_12May_named) <- "RNA"
```


## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E14.5_12May_named <- subset(fl_samples_E14.5_12May_named, idents =c("EP1a", "EP1b", "EP2a", "EP2b", "EP3a", "EP4a", "EP5a", "MEP", "PEBa", "PEBb", "PEBc", "Mk", "Hepato"))

```



```{r}
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named <- SCTransform(fl_samples_E14.5_12May_named,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```





```{r, eval=FALSE}
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers_fl_samples_E14.5_12May_named <- FindAllMarkers(object = fl_samples_E14.5_12May_named, 
                          only.pos = TRUE,
                          assay = "SCT",
                          logfc.threshold = 0.25,
                          min.pct = 0.25,
                          thresh.use = 0.25)
```


```{r, eval=FALSE} 
# Arrange markers by average log2fold change
markers_fl_samples_E14.5_12May_named <- markers_fl_samples_E14.5_12May_named %>% 
  dplyr::group_by(cluster) %>% 
  dplyr::arrange(desc(avg_log2FC), .by_group = TRUE)

```


```{r}
# Let us look at the top markers of each cluster

top_3_markers_fl_samples_E14.5_12May_named <- markers_fl_samples_E14.5_12May_named %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 3)

#top_30_markers_fl_samples_E14.5_12May <- markers_fl_samples_E14.5_12May %>% group_by(cluster) %>% 
  #slice_max(avg_log2FC, n = 30)
```


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)
```



```{r}
# Set colors
clust_cols <- c(EP1a = "#16FF32", EP1b =  "#565656", EP2a = "#FE00FA", EP2b = "#DEA0FD", EP3a = "#1C8356", EP4a = "darkorange2", EP5a = "#325A9B", MEP = "#F6222E", Mk = "#00AFBB", PEBa = "#3283FE", PEBb = "#1CFFCE", PEBc = "goldenrod1", Hepato = "#F8A19F")
```
Ery4b = "#2ED9FF",Ery5b = "#90AD1C", "#85660D",
 Ery3b = "#2ED9FF",EP5a = "#325A9B",  


```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named,
        reduction = "umap",
        cols = clust_cols,
        label = TRUE,
        label.size = 11,
        pt.size = 1.5) + NoLegend() +
            theme(axis.title.x = element_text(size =50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
theme(axis.text = element_text(size = 50, face = "bold"), 
        panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 1)) 
ggsave("repeated_umap2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
```


```{r}
fl_samples_E14.5_12May_named@active.ident <- factor(fl_samples_E14.5_12May_named@active.ident, levels = c("MEP", "Mk", "EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "PEBa", "EP1b", "EP2b","PEBb", "PEBc", "Hepato"))
```


```{r, eval=FALSE}
fl_samples_E14.5_12May_named$cluster_ident <- factor(fl_samples_E14.5_12May_named$cluster_ident, levels = c("MEP", "Mk", "EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "PEBa", "EP1b", "EP2b","PEBb", "PEBc", "Hepato"))
```



```{r}
(ncells_E14.5 <- data.frame(table(fl_samples_E14.5_12May_named$cluster_ident)) %>% dplyr::rename("Clusters" = Var1))

(ncells_E14.5 <- mutate(ncells_E14.5, percent = round((Freq/sum(Freq))*100, digits = 2)))
ncells_E14.5$percent <- paste0(ncells_E14.5$percent,"%")
ncells_E14.5

ggplot(ncells_E14.5, aes(x = Clusters, y = Freq, fill = Clusters)) +
  geom_col() +
  scale_fill_manual(values = c(EP1a = "#16FF32", EP1b =  "#565656", EP2a = "#FE00FA", EP2b = "#DEA0FD", EP3a = "#1C8356", EP4a = "darkorange2", EP5a = "#325A9B", MEP = "#F6222E", Mk = "#00AFBB", PEBa = "#3283FE", PEBb = "#1CFFCE", PEBc = "goldenrod1", Hepato = "#F8A19F")) +
  theme(axis.text = element_text(size = 50),
        panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) +
        theme(axis.title.x = element_text(size =50, face = "bold")) +
        theme(axis.title.y = element_text(size = 50, face = "bold")) +
  geom_text(aes(label = percent), vjust = 0.5, hjust = -0.1, size = 11, face = "bold") +
  #scale_y_continuous(breaks = seq(0, 2000, by=1000), limits=c(0,2000)) +
  scale_y_continuous(breaks = seq(0, 2000, by=1000), limits=c(0,2000)) +
  xlab("Cell types") + 
 # theme_bw() + 
  NoLegend() +
  coord_flip()

ggsave("repeated_cellsincluster_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
DefaultAssay(fl_samples_E14.5_12May_named) <- "RNA"

# First, subset object into replicates
E14.5_rep1 <- subset(x = fl_samples_E14.5_12May_named, subset = sample == "FL_E14.5_rep_1")
E14.5_rep2 <- subset(x = fl_samples_E14.5_12May_named, subset = sample == "FL_E14.5_rep_2")

# Average genes per cell in each object
(ave_genes_percell_e14.5_rep1 <- mean(Matrix::colSums(GetAssayData(E14.5_rep1, slot = 'counts') > 0)))
(ave_genes_percell_e14.5_rep2 <- mean(Matrix::colSums(GetAssayData(E14.5_rep2, slot = 'counts') > 0)))

# Get average reads/counts per cell
(ave_counts_percell_e14.5_rep1 <- mean(E14.5_rep1@meta.data$nUMI))
(ave_counts_percell_e14.5_rep2 <- mean(E14.5_rep2@meta.data$nUMI))

rm(E14.5_rep1, E14.5_rep2)
```
[1] 4083.514
[1] 4096.327
[1] 17409.66
[1] 17697.87



```{r}
DefaultAssay(fl_samples_E14.5_12May_named) <- "SCT"

# Megakaryocyte progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("F2r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Itga2b"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Gp1bb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
 
a
ggsave("R_Mk_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
b
ggsave("R_Mk_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
c
ggsave("R_Mk_marker3_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
```



```{r}
# Megakaryocytes
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Myl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Vwf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Ppbp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

d <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Alox12"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

e <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Zyx"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_Mk_marker4_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
b
ggsave("R_Mk_marker5_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
c
ggsave("R_Mk_marker6_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
d
ggsave("R_Mk_marker8_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
e
ggsave("R_Mk_marker7_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 500)
```


```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Gata2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Klf1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_MEP_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
b
ggsave("R_MEP_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
c
ggsave("R_MEP_marker3_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
```




```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Epor"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Car1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Muc13"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
 
a
ggsave("R_MEP_marker4_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
b
ggsave("R_MEP_marker5_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
c
ggsave("R_MEP_marker6_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
```



```{r}
# Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Myb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

d <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

a
ggsave("R_EryP_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
b
ggsave("R_EryP_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
c
ggsave("R_EryP_marker3_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
d
ggsave("R_EryP_marker4_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
```

```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Hbb-bs"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Hbb-bt"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

c <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Gypa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

d <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Hba-a1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

e <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 1)
f <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            pt.size = 1)

a
ggsave("R_Erythb_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
b
ggsave("R_Erythb_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
c
ggsave("R_Erythb_marker3_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
d
ggsave("R_Erythb_marker4_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff", dpi = 300)
e
ggsave("R_Erythb_marker5_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
f
ggsave("R_Erythb_marker6_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


```{r}
# Hepatoblasts
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Afp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Alb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_Hepato_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Hepato_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



# LOOK AT SOME Y AND X CHROMOSOME GENES

```{r}
# Male genes
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Eif2s3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Ddx3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_Male_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Male_marker2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Female genes
a <- FeaturePlot(fl_samples_E14.5_12May_named, 
            reduction = "umap", 
            features = c("Xist"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

a
ggsave("R_Female_marker1_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
# Plot top 3 markers per object 
a <- DotPlot(
  fl_samples_E14.5_12May_named,
  assay = NULL,
  features = unique(top_3_markers_fl_samples_E14.5_12May_named$gene),
  cols = c("lightgrey", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + RotatedAxis() + 
  theme(axis.text = element_text(size = 40, face = "bold")) + 
              theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
  theme(legend.text = element_text(size = 20, face = "bold")) +
  theme(plot.title = element_text(face = "bold", size = 30)) + 
  theme(legend.title = element_text(size = 20), legend.position = "top") +
  ggtitle("FL_E14.5 Top 3 marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")



known_markers1 <- c("Gypa", "Tfrc", "Gata1", "Kit","Hba-x", "Hbb-y", "Gata2", "Afp", "Alb", "Hnf4a", "Pf4", "Vwf", "Itga2b", "Thbs1", "Gata2", "Gata1", "Klf1", "Car1", "Ppbp", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Gp1bb", "Cd24a", "Muc13", "Myb", "Epor", "F2r", "Myl9", "Pf4")
   

  
  
known_markers1  <- markers_fl_samples_E14.5_12May_named %>% 
                dplyr::filter(gene %in% known_markers1)

b <- DotPlot(
  fl_samples_E14.5_12May_named,
  assay = NULL,
  features = unique(known_markers1$gene),
  cols = c("darkblue", "red"),
  col.min = -2.5,
  col.max = 2.5,
  dot.min = 0,
  dot.scale = 6,
  idents = NULL,
  group.by = NULL,
  split.by = NULL,
  cluster.idents = FALSE,
  scale = TRUE,
  scale.by = "radius",
  scale.min = NA,
  scale.max = NA
) + 
  RotatedAxis() +
 theme(axis.text = element_text(size = 40, face = "bold")) + 
  theme(axis.text.x = element_text(angle = 80)) +
              theme(axis.title.x = element_text(size = 40, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
  theme(legend.text = element_text(size = 20, face = "bold")) +
  theme(plot.title = element_text(face = "bold", size = 30)) + 
  theme(legend.title = element_text(size = 20), legend.position = "top") + 
  ggtitle("FL_E14.5 - Known marker genes") + scale_colour_gradient2(low="darkblue", mid="lightgrey", high="red")

a
ggsave("R_Dotplot_marker1_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "pdf", dpi = 300)
b
ggsave("R_Dotplot_marker2_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "pdf", dpi = 300)
```



```{r}
# Plot UMAP
set.seed(1203)
DimPlot(object = fl_samples_E14.5_12May_named,
        reduction = "umap",
        cols = clust_cols,
        split.by = "sample",
        label.size = 6,
        pt.size = 1.5)
ggsave("R_replicates_color_umap_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named,
        reduction = "umap",
        label = FALSE,
        group.by = "sample",
        label.size = 6,
        pt.size = 1.5) 
ggsave("R_by_sample_umap_of_replicates_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


# Plot top 3 genes
```{r}
top_3_markers <- top_3_markers_fl_samples_E14.5_12May_named %>% 
                  ungroup() %>%
                  dplyr::select(gene) %>% 
                  unique() %>% 
                  .[!is.na(.)]

mat <- fl_samples_E14.5_12May_named@assays$SCT@data[top_3_markers, ] %>% as.matrix()
mat <- t(scale(t(mat)))

cols_sample_group <- c(FL_E14.5 = "#E2E2E2" )

clust <- c("MEP", "Mk", "EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "PEBa", "EP1b", "EP2b","PEBb", "PEBc", "Hepato")

clusters <- fl_samples_E14.5_12May_named$cluster_ident

                
annotation_col <-  data.frame(E14.5=clusters)

annotation_colors <- list(E14.5 = clust_cols,
                          Timepoint=cols_sample_group)
                                        
# Determine expression ranges in the matrix
quantile(mat, c(0.1, 0.95))
cols <- circlize::colorRamp2(c(-1, 0, 2), c("blue", "black", "yellow"))


#pdf(file="~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots/R_top3_heatmap_fl_samples_E14.5_12May_named.pdf", width = 25, height = 25)
heat <- ComplexHeatmap::Heatmap(mat, name="E14.5 Top3",
                        column_split = factor(clusters, levels = clust),
                        cluster_columns = TRUE,
                        show_column_dend = FALSE,
                        show_column_names = FALSE,
                        column_title = clust,
                        cluster_column_slices = FALSE,
                        column_title_gp = gpar(fontsize = 20),
                        row_names_gp = gpar(fontsize = 25),
                        column_gap = unit(0.5, "mm"),
                        cluster_rows = TRUE,
                        show_row_dend = FALSE,
                        col = cols,
                        row_gap = unit(10, "mm"),
                        column_title_rot = 60,
                        top_annotation = HeatmapAnnotation(df = annotation_col, col = annotation_colors, show_legend = FALSE, show_annotation_name = FALSE),
                        heatmap_legend_param = list(labels_gp = gpar(fontsize = 30), title_gp = gpar(fontsize = 30, fontface = "bold")),
                        width = 25,
                        height = 25,
                        use_raster = TRUE,
                        raster_quality = 4)
draw(heat)
#dev.off()
#annotation_legend_param = list(labels_gp = gpar(fontsize = 30),show_annotation_name = TRUE,title_gp = gpar(fontsize = 30, fontface = "bold")

```


#Code copied from [https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/]
```{r}
## remove the x-axis text and tick
## plot.margin to adjust the white space between each plot.
## ... pass any arguments to VlnPlot in Seurat
modify_vlnplot<- function(obj, 
                          feature, 
                          pt.size = 0, 
                          plot.margin = unit(c(-0.75, 0, -0.75, 0), "cm"),
                          ...) {
  p<- VlnPlot(obj, features = feature, pt.size = pt.size, ... )  + 
    xlab("") + ylab(feature) + ggtitle("") + 
    theme(legend.position = "none", 
          axis.text.x = element_blank(),
          plot.title = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank(), 
          axis.title.y = element_text(size = rel(1), angle = 0, face = "bold"), 
          axis.text.y = element_text(size = rel(1)), 
          plot.margin = plot.margin ) + geom_boxplot(outlier.shape = NA) 
  return(p)
}

## extract the max value of the y axis
extract_max<- function(p){
  ymax<- max(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
  return(ceiling(ymax))
}


## main function
StackedVlnPlot<- function(obj, features,
                          pt.size = 0, 
                          plot.margin = unit(c(-0.75, 0, -0.75, 0), "cm"),
                          ...) {
  
  plot_list<- purrr::map(features, function(x) modify_vlnplot(obj = obj,feature = x, ...))
  
  # Add back x-axis title to bottom plot. patchwork is going to support this?
  plot_list[[length(plot_list)]]<- plot_list[[length(plot_list)]] +
    theme(axis.text.x=element_text(angle = 45, face = "bold"), axis.ticks.x = element_line())
  
  # change the y-axis tick to only max value 
  ymaxs<- purrr::map_dbl(plot_list, extract_max)
  plot_list<- purrr::map2(plot_list, ymaxs, function(x,y) x + 
                            scale_y_continuous(breaks = c(y)) + 
                            expand_limits(y = y))

  p<- patchwork::wrap_plots(plotlist = plot_list, ncol = 1)
  return(p)
}
```



```{r}
# MEP
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Gata1", "Car1", "Gata2", "Klf1", "Muc13", "Epor"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("MEP_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


```{r}
# Megakaryocytes
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Pf4", "Itga2b", "Ppbp", "Thbs1", "Gp9", "Zyx"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("Mk_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```
Zyx has the highest expression in megakaryocytes. It could be a new marker gene for identifying megakaryocytes.


```{r}
# Erythroid cells
StackedVlnPlot(obj = fl_samples_E14.5_12May_regressed_named, features = c("Myb", "Kit", "Gata1", "Tfrc", "Klf1", "Gypa", "Cd24a", "Epor", "Hba-a1", "Hba-a2", "Hbb-bs", "Hbb-bt", "Hbb-bh1"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("All_Ery_violin_fl_samples_E14.5_12May_regressed_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

# Look at Cd24a and Kit expression levels in the erythroid clusters
```{r}
# Erythroid cells
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Kit", "Tfrc", "Klf1", "Cd24a"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data") 

ggsave("Cd24a_Kit_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```





```{r}
# Hepatoblasts
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Afp", "Alb", "Ttr", "Hnf4a"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Hepato_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

```{r}
# Endothelial
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Cldn5", "Cdh5"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Endo_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```

```{r}
# Mesenchymal cells
StackedVlnPlot(obj = fl_samples_E14.5_12May_named, features = c("Alcam", "Col1a2"), group.by = "cluster_ident", add.noise = FALSE, pt.size = 0, cols = clust_cols, slot = "data")

ggsave("Mesench_violin_fl_samples_E14.5_12May_named.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```



# Use differentially expressed genes to find new possible marker genes for the clusters


## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E14.5_12May_named_subset <- subset(fl_samples_E14.5_12May_named, idents =c("MEP", "Mk", "EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "PEBa", "EP1b", "EP2b","PEBb", "PEBc"))
```



```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset) <- "RNA"
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset <- SCTransform(fl_samples_E14.5_12May_named_subset,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```




```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named_subset,
        reduction = "umap",
        label = FALSE,
        group.by = "Phase",
        label.size = 6,
        pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size =50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold")) +
            theme(legend.text = element_text(size = 50, face = "bold")) +
            guides(color = guide_legend(override.aes = list(size = 5))) +
            theme(axis.text = element_text(size = 50, face = "bold"), 
            panel.background = element_rect(fill = "white"), 
            plot.background = element_rect(fill = "white", colour = "black", linewidth = 1)) 
ggsave("repeated_phase_umap2_fl_samples_E14.5_12May_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Set colors
clust_cols_subset <- c(EP1a = "#16FF32", EP1b =  "#565656", EP2a = "#FE00FA", EP2b = "#DEA0FD", EP3a = "#1C8356", EP4a = "darkorange2", EP5a = "#325A9B", MEP = "#F6222E", Mk = "#00AFBB", PEBa = "#3283FE", PEBb = "#1CFFCE", PEBc = "goldenrod1")
```
EP5a = "#325A9B", 


```{r}
# Plot UMAP
set.seed(12098)
DimPlot(object = fl_samples_E14.5_12May_named_subset,
        reduction = "umap",
        cols = clust_cols_subset,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend()

ggsave("R_umap_fl_samples_E14.5_12May_named_subset.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```




I WILL DO TRAJECTORY INFERENCE USING CELLRANK AND NOT PALANTIR (CELLRANK WORKS BETTER FOR THIS DATA) BEFORE I COME BACK TO DIFFERENTIAL GENE EXPRESSION ANALYSIS AND PATHWAY ANALYSIS

```{r}
#dir.create("~/myproject/sc_analysis_project/msc_project/rnavelocity/E14.5_rnavelocity")
E14.5_rnavelo <- "~/myproject/sc_analysis_project/msc_project/rnavelocity/E14.5_rnavelocity/"
```


## PREPARE ERYTHROID SEURAT OBJECT METADATA FOR RNA VELOCITY (https://smorabit.github.io/tutorials/8_velocyto/)

## Ery
```{r}
# Make a copy of the original object to avoid stories that touch
fl_samples_E14.5_12May_named_subset1 <- fl_samples_E14.5_12May_named_subset
# save metadata table
fl_samples_E14.5_12May_named_subset1$barcode <- colnames(fl_samples_E14.5_12May_named_subset1)
fl_samples_E14.5_12May_named_subset1$UMAP_1 <- fl_samples_E14.5_12May_named_subset1@reductions$umap@cell.embeddings[,1]
fl_samples_E14.5_12May_named_subset1$UMAP_2 <- fl_samples_E14.5_12May_named_subset1@reductions$umap@cell.embeddings[,2]
```  


```{r}


# Select column variables of interest
Ery_E14.5_metadata <- fl_samples_E14.5_12May_named_subset1@meta.data
Ery_E14.5_metadata <- Ery_E14.5_metadata %>% dplyr::select(c(seq_folder, cells, sample, sample_group, sample_origin, S.Score, G2M.Score, Phase, cluster_ident, barcode, UMAP_1, UMAP_2)) %>% dplyr::rename("CellID" = cells)

fl_samples_E14.5_12May_named_subset1@meta.data <- Ery_E14.5_metadata
```

```{r}
write.csv(fl_samples_E14.5_12May_named_subset1@meta.data, file = paste0(E14.5_rnavelo, "Ery_E14.5_metadata.csv"), quote = F, row.names = F)

```


```{r}
# Write expression counts matrix
Ery_E14.5_counts_matrix <- GetAssayData(fl_samples_E14.5_12May_named_subset1, assay = "SCT", slot = "counts")
writeMM(Ery_E14.5_counts_matrix, file = paste0(E14.5_rnavelo, "Ery_E14.5_counts.mtx"))
```

```{r}
# Write dimensionality reduction matrix
write.csv(fl_samples_E14.5_12May_named_subset1@reductions$pca@cell.embeddings, file = paste0(E14.5_rnavelo, "Ery_E14.5_pca.csv"), quote = F, row.names = F)
```


```{r}
# Write gene names
write.table(data.frame("gene" = rownames(Ery_E14.5_counts_matrix)), file = paste0(E14.5_rnavelo, "Ery_E14.5_gene_names.csv"), quote = F, row.names = F, col.names = F)
```




## Subset Erythroid progenitors 

## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E14.5_12May_named_subset_DEG <- subset(fl_samples_E14.5_12May_named_subset, idents =c("EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "EP1b", "EP2b"))
 
```



```{r}
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset_DEG <- SCTransform(fl_samples_E14.5_12May_named_subset_DEG,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



## Reannotate clusters in FL_E14.5 samples for DEG analysis between erythroid progenitors

```{r}
# Fetal liver
fl_samples_E14.5_12May_named_subset_DEG <-  RenameIdents(object = fl_samples_E14.5_12May_named_subset_DEG,
                                                  "EP1a" = "EPa",
                                                  "EP1b" = "EPb",
                                                  "EP2a" = "EPa",
                                                  "EP2b" = "EPb",
                                                  "EP3a" = "EPa",
                                                  "EP4a" = "EPa",
                                                  "EP5a" = "EPa")

fl_samples_E14.5_12May_named_subset_DEG$cluster_ident <- Idents(fl_samples_E14.5_12May_named_subset_DEG)

```


```{r}
table(fl_samples_E14.5_12May_named_subset_DEG$cluster_ident, fl_samples_E14.5_12May_named_subset_DEG$sample_group)
```


```{r}
# Set colors
clust_cols_EP <- c(EPa = "#16FF32", EPb = "goldenrod1")
```





```{r}
# Plot UMAP
set.seed(12098)
DimPlot(object = fl_samples_E14.5_12May_named_subset_DEG,
        reduction = "umap",
        cols = clust_cols_EP,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend()

ggsave("R_umap_fl_samples_E14.5_12May_named_subset_DEG.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```



# DIFFERENTIAL GENE EXPRESSION BETWEEN EPa and EPb

```{r}
EPa_vs_EPb_E14.5 <- FindMarkers(fl_samples_E14.5_12May_named_subset_DEG, 
                                  ident.1 = "EPa",
                                  ident.2 = "EPb",
                                  assay = "SCT",
                                  min.pct = 0.3)

EPa_vs_EPb_E14.5
```


```{r}
# Arrange adjusted p-values
EPa_vs_EPb_E14.5 <- EPa_vs_EPb_E14.5 %>% 
  dplyr::arrange("p_val_adj") %>% 
  dplyr::mutate(gene = rownames(EPa_vs_EPb_E14.5))

# What is the total number of genes in the DE analysis?
nrow(EPa_vs_EPb_E14.5)

# How many genes have significant adjusted p-values,  that is differentially expressed?
EPa_vs_EPb_E14.5_signif <- EPa_vs_EPb_E14.5 %>% 
                              dplyr::filter(p_val_adj < 0.05)
nrow(EPa_vs_EPb_E14.5_signif)

# How many genes are upregulated?
(EPa_vs_EPb_E14.5_upreg <- EPa_vs_EPb_E14.5_signif %>% 
                              dplyr::filter(avg_log2FC > 0))
nrow(EPa_vs_EPb_E14.5_upreg)

# How many genes are down-regulated?
(EPa_vs_EPb_E14.5_downreg <- EPa_vs_EPb_E14.5_signif %>% 
                              dplyr::filter(avg_log2FC < 0))
nrow(EPa_vs_EPb_E14.5_downreg)
```


```{r}
EnhancedVolcano(EPa_vs_EPb_E14.5_signif, 
                x = "avg_log2FC", 
                y = "p_val_adj", 
                xlim = c(min(EPa_vs_EPb_E14.5_signif[["avg_log2FC"]], na.rm = TRUE) - 0.5, max(EPa_vs_EPb_E14.5_signif[["avg_log2FC"]], na.rm = TRUE) + 0.5),
                lab = rownames(EPa_vs_EPb_E14.5_signif),
                pCutoff = 10e-5, 
                FCcutoff = 0,
                col = c("grey30", "forestgreen", "dodgerblue4", "red4"),
                title = "", 
                ylab = "-log10(p-val adj)", 
                subtitle = "E14.5 EPa vs EPb", 
                legendLabels = c("Not sig.", expression(Log[2] ~ FC), "p-value", expression(p - value ~ +
                ~ log[2] ~ FC)))

ggsave("EPa.EPb_volcano_fl_samples_E14.5_12May_named_subset_DEG.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/",
       width = 15, height = 15)
```



```{r}
#dir.create("/home/BCCRC.CA/eamutaigwe/myproject/sc_analysis_project/msc_project/DE_analysis/results/FL_E14.5")
# Save the DEGs (all genes, not just the significant ones)
saveRDS(EPa_vs_EPb_E14.5,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5.rds")))


# Save significant DEGs
saveRDS(EPa_vs_EPb_E14.5_signif,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5_signif.rds")))

# Save upregulated genes
saveRDS(EPa_vs_EPb_E14.5_upreg,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5_upreg.rds")))
write.csv(EPa_vs_EPb_E14.5_upreg,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5_upreg.csv")))

# Save downregulated genes
saveRDS(EPa_vs_EPb_E14.5_downreg,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5_downreg.rds")))
write.csv(EPa_vs_EPb_E14.5_downreg,
          paste0(here("DE_analysis/results/FL_E14.5", "EPa_vs_EPb_E14.5_downreg.csv")))
```



# GO ANALYSIS OF DIFFERENTIALLY EXPRESSED ERYTHROID AND MEGAKARYOCYTE GENES. I did pathway enrichment instead

```{r}
# Load libraries
library(clusterProfiler)
library(msigdbr)
library(GOplot)
```



```{r}
# Extract databases of interest
msigdbr_df1 <- msigdbr(species = "mouse", category = "C2", subcategory = "CP:REACTOME")
msigdbr_df2 <- msigdbr(species = "mouse", category = "C2", subcategory = "CP:KEGG")
msigdbr_df3 <- msigdbr(species = "mouse", category = "C5", subcategory = "GO:BP")
msigdbr_df4 <- msigdbr(species = "mouse", category = "C5", subcategory = "GO:CC")
msigdbr_df5 <- msigdbr(species = "mouse", category = "C5", subcategory = "GO:MF")
msigdbr_df6 <- msigdbr(species = "mouse", category = "H")


# Extract columns of interest(gene set name and gene symbol)
# Reactome Pathways
msigdbr_t2g1 <- msigdbr_df1 %>% dplyr::distinct(gs_name, gene_symbol) %>% as.data.frame()
# KEGG Pathways
msigdbr_t2g2 <- msigdbr_df2 %>% dplyr::distinct(gs_name, gene_symbol) %>% as.data.frame()
# Biological Processes
msigdbr_t2g3 <- msigdbr_df3 %>% dplyr::distinct(gs_name, gene_symbol) %>% as.data.frame()
```



```{r}
# ENRICHER
# Up_regulated genes
EPa_vs_EPb_E14.5_upreg_Reactome <- enricher(gene = EPa_vs_EPb_E14.5_upreg$gene, TERM2GENE = msigdbr_t2g1) 
EPa_vs_EPb_E14.5_upreg_Reactome@result$Description <- EPa_vs_EPb_E14.5_upreg_Reactome@result$Description %>% str_replace_all("REACTOME_", "")

EPa_vs_EPb_E14.5_upreg_BP <- enricher(gene = EPa_vs_EPb_E14.5_upreg$gene, TERM2GENE = msigdbr_t2g3) 
EPa_vs_EPb_E14.5_upreg_BP@result$Description <- EPa_vs_EPb_E14.5_upreg_BP@result$Description %>% str_replace_all("GOBP_", "")
```



# PLOT ERYTHROID DIFFERENTIALLY EXPRESSED GENES
```{r}
# Select terms of interest for up-regulated genes (top 10 significant terms)

# Plot up-regulated genes
#pdf(file= "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/Ery1a_vs_MEP_E14.5_upreg_Reactome.pdf", width = 15, height = 15)
a <- barplot(EPa_vs_EPb_E14.5_upreg_Reactome, x = "Count", color = "p.adjust", showCategory = 10, font.size = 14, title = "REACTOME sig.terms - upreg. genes in E14.5 EPa vs EPb", label_format = 30)
a
#dev.off()


#pdf(file= "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/Ery1a_vs_MEP_E14.5_upreg_BP.pdf", width = 15, height = 15)
a <- barplot(EPa_vs_EPb_E14.5_upreg_BP, x = "Count", color = "p.adjust", showCategory = 10, font.size = 14, title = "BP sig.terms - upreg. genes in E14.5 EPa vs EPb", label_format = 30)
a
#dev.off()
```


```{r}
# ENRICHER
# Down_regulated genes
EPa_vs_EPb_E14.5_downreg_Reactome <- enricher(gene = EPa_vs_EPb_E14.5_downreg$gene, TERM2GENE = msigdbr_t2g1)
EPa_vs_EPb_E14.5_downreg_Reactome@result$Description <- EPa_vs_EPb_E14.5_downreg_Reactome@result$Description %>% str_replace_all("REACTOME_", "")


EPa_vs_EPb_E14.5_downreg_BP <- enricher(gene = EPa_vs_EPb_E14.5_downreg$gene, TERM2GENE = msigdbr_t2g3)
EPa_vs_EPb_E14.5_downreg_BP@result$Description <- EPa_vs_EPb_E14.5_downreg_BP@result$Description %>% str_replace_all("GOBP_", "")
```


# PLOT ERYTHROID DIFFERENTIALLY EXPRESSED GENES
```{r}
# Select terms of interest for down-regulated genes (top 10 significant terms)

# Plot down-regulated genes
#pdf(file= "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/Ery1a_vs_MEP_E14.5__downreg_Reactome.pdf", width = 15, height = 15)
a <- barplot(EPa_vs_EPb_E14.5_downreg_Reactome, x = "Count", color = "p.adjust", showCategory = 10, font.size = 14, title = "REACTOME sig.terms - downreg. genes in E14.5 EPa vs EPb", label_format = 30)
a
#dev.off()

#pdf(file= "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/Ery1a_vs_MEP_E14.5__downreg_BP.pdf", width = 15, height = 15)
a <- barplot(EPa_vs_EPb_E14.5_downreg_BP, x = "Count", color = "p.adjust", showCategory = 10, font.size = 14, title = "BP sig.terms-downreg. genes in E14.5 EPa vs EPb", label_format = 30)
a
#dev.off()
```




## Reannotate clusters in FL_E14.5 samples for pseudobulk DEG analysis

```{r}
# Fetal liver
fl_samples_E14.5_12May_named_subset_DEG_merge <-  RenameIdents(object = fl_samples_E14.5_12May_named_subset_DEG,
                                                  "EPa" = "EP",
                                                  "EPb" = "EP")

fl_samples_E14.5_12May_named_subset_DEG_merge$cluster_ident <- Idents(fl_samples_E14.5_12May_named_subset_DEG_merge)

```


```{r}
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset_DEG_merge <- SCTransform(fl_samples_E14.5_12May_named_subset_DEG_merge,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```




# LOOK AT THE CLUSTERING AND DGE AGAIN AFTER REGRESSING OUT SEX


```{r}
# Rename original seurat object to prevent tampering with it
fl_samples_E14.5_12May_named_subset_nosexdoublets <- fl_samples_E14.5_12May_named_subset
```




# Determine sex of cells at E14.5 using cellXY
```{r}
library(cellXY)
library(speckle)
library(SingleCellExperiment)
library(CellBench)
library(org.Mm.eg.db)
library(caret)
library(randomForest)

fl_samples_E14.5_12May_named_subset_nosexdoublets_counts <- fl_samples_E14.5_12May_named_subset_nosexdoublets@assays$RNA@counts
ann <- AnnotationDbi::select(org.Mm.eg.db, keys=rownames(fl_samples_E14.5_12May_named_subset_nosexdoublets),
              columns=c("ENSEMBL","SYMBOL"), keytype="SYMBOL")
```


```{r}
# How many Ensembl identifiers have an associated gene symbol
which(is.na(ann$ENSEMBL)) %>% length()

# How many of them are unique
which(duplicated(ann$ENSEMBL)) %>% length()

# Determine the indices for the non-NA genes
non_na_idx <- !is.na(ann$ENSEMBL)
non_duplicated_idx <- which(duplicated(ann$ENSEMBL) == FALSE)

# How many rows does annotations have?
ann %>% nrow()

# Return only the non-duplicated genes using the indices
ann <- ann[non_na_idx, ]
ann <- ann[non_duplicated_idx, ]

# How many rows does annotations now have?
ann %>% nrow()


m <- match(rownames(fl_samples_E14.5_12May_named_subset_nosexdoublets_counts), ann$SYMBOL)
rownames(fl_samples_E14.5_12May_named_subset_nosexdoublets_counts) <- ann$SYMBOL[m]
```


```{r}
db_pred =findMfDoublet(x=fl_samples_E14.5_12May_named_subset_nosexdoublets_counts, genome="Mm")
db_pred$cluster_ident <- Idents(fl_samples_E14.5_12May_named_subset_nosexdoublets)
print(table(db_pred$prediction, db_pred$cluster_ident))
```

Out of 6991 cells in the fl_samples_E14.5_12May_named_subset seurat object, 94 were predicted to be sex doublets and were filtered out. 6897 cells remaining.

```{r}
# Filter out sex doublets
singlets = colnames(fl_samples_E14.5_12May_named_subset_nosexdoublets_counts)[db_pred$prediction=="Singlet"]
length(singlets)
fl_samples_E14.5_12May_named_subset_nosexdoublets <- subset(fl_samples_E14.5_12May_named_subset_nosexdoublets, cells = singlets)
fl_samples_E14.5_12May_named_subset_nosexdoublets[]
#dir.create("~/myproject/sc_analysis_project/msc_project/DE_analysis/results/FL_E14.5/EPa_vs_EPb_upregulated")
#dir.create("~/myproject/sc_analysis_project/msc_project/DE_analysis/results/FL_E14.5/EPa_vs_EPb_downregulated")
```

```{r}
# SCTransform the regressed and subset singlet object
fl_samples_E14.5_12May_named_subset_nosexdoublets <- SCTransform(fl_samples_E14.5_12May_named_subset_nosexdoublets,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)

```


```{r}
FeaturePlot(fl_samples_E14.5_12May_named_subset_nosexdoublets, 
            reduction = "umap", 
            features = c("Xist"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
```


```{r}
fl_samples_E14.5_12May_named_subset_nosexdoublets_counts <- fl_samples_E14.5_12May_named_subset_nosexdoublets@assays$RNA@counts
sex <- classifySex(fl_samples_E14.5_12May_named_subset_nosexdoublets_counts, genome="Mm")
table(sex$prediction)
```

Female   Male     NA 
  4241   2414    241 

  
```{r}
sex <- rownames_to_column(sex, var = "cells")
metadata <- fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data
metadata <- left_join(metadata, sex, by = "cells")

na_sex <- dplyr::filter(metadata, prediction == "NA")
nrow(na_sex)

fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data <- metadata
rownames(fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data) <- fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data$cells
table(fl_samples_E14.5_12May_named_subset_nosexdoublets$sample_group, fl_samples_E14.5_12May_named_subset_nosexdoublets$prediction)
fl_samples_E14.5_12May_named_subset_nosexdoublets$prediction <- factor(fl_samples_E14.5_12May_named_subset_nosexdoublets$prediction)
```
          Female Male   NA
  FL_E14.5   4241 2414  241


```{r}
fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data <- fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data %>% 
  mutate(
    sex = case_when(
  prediction == "Female" ~ "Female", 
  prediction == "Male" ~ "Male",
  prediction == "NA" ~ "unknown"
))

fl_samples_E14.5_12May_named_subset_nosexdoublets$sex <- factor(fl_samples_E14.5_12May_named_subset_nosexdoublets$sex)  
  
table(fl_samples_E14.5_12May_named_subset_nosexdoublets$sample_group, fl_samples_E14.5_12May_named_subset_nosexdoublets$sex)
```

           Female Male unknown
  FL_E14.5   4241 2414     241



```{r}
#str(fl_samples_E14.5_12May_named_subset_regressed)
head(fl_samples_E14.5_12May_named_subset_nosexdoublets@meta.data)

```

```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named_subset_nosexdoublets,
        reduction = "umap",
        label = TRUE,
        label.size = 6,
        pt.size = 1.5)


```

```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_nosexdoublets) <- "RNA"
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset_nosexdoublets <- SCTransform(fl_samples_E14.5_12May_named_subset_nosexdoublets,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score", "sex"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_nosexdoublets) <- "SCT"
# Female gene
FeaturePlot(fl_samples_E14.5_12May_named_subset_nosexdoublets, 
            reduction = "umap", 
            features = c("Xist"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

ggsave("R_Female_marker1_fl_samples_E14.5_12May_named_subset_nosexdoublets.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Male genes
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_nosexdoublets, 
            reduction = "umap", 
            features = c("Eif2s3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_nosexdoublets, 
            reduction = "umap", 
            features = c("Ddx3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1) +
            theme(axis.text.x = element_text(face = "bold", size = 50)) +
            theme(axis.text.y = element_text(face = "bold", size = 50)) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.title.x = element_text(size = 50, face = "bold")) +
            theme(axis.title.y = element_text(size = 50, face = "bold"))
a
ggsave("R_Male_marker1_fl_samples_E14.5_12May_named_subset_nosexdoublets.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("R_Male_marker2_fl_samples_E14.5_12May_named_subset_nosexdoublets.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```

# REGRESS OUT SEX GENES AND REDO DGE ANALYSIS

```{r}
sex_genes <- list(c("Med14", "Usp9x", "Ddx3x", "Maoa","Fundc1","Kdm6a","Uba1","Cdk16","Plxnb3","L1cam","Arhgap4", "Naa10","Renbp","Hcfc1","Ikbkg","Gab3","Tbl1x","Prkx","Zfx","Eif2s3x", "Tsix","Xist","Sh3bgrl", "Chm","Pcdh11x", "Nap1l3",  "Srpx2", "Sytl4","Morf4l2", "Alg13","Ribc1","Smc1a","Kdm5c","Gpr143",  "Eif1ax",  "Rbbp7",   "Syap1",   "S100g", "Grpr","Car5b","Pir","Gemin8","Gpm6b","Ofd1","Rab9","Tmsb4x",  "Amelx","Clcn4","Dazl","Kdm5d","Eif2s3y", "Uty", "Ddx3y", "Usp9y")) 

sex_genes_vector <- c("Med14", "Usp9x", "Ddx3x", "Maoa","Fundc1","Kdm6a","Uba1","Cdk16","Plxnb3","L1cam","Arhgap4", "Naa10","Renbp","Hcfc1","Ikbkg","Gab3","Tbl1x","Prkx","Zfx","Eif2s3x", "Tsix","Xist","Sh3bgrl", "Chm","Pcdh11x", "Nap1l3",  "Srpx2", "Sytl4","Morf4l2", "Alg13","Ribc1","Smc1a","Kdm5c","Gpr143",  "Eif1ax",  "Rbbp7",   "Syap1",   "S100g", "Grpr","Car5b","Pir","Gemin8","Gpm6b","Ofd1","Rab9","Tmsb4x",  "Amelx","Clcn4","Dazl","Kdm5d","Eif2s3y", "Uty", "Ddx3y", "Usp9y")

# Incomplete list
#sex_genes <- c("Med14", "Usp9x", "Ddx3x",  "Maoa", "Fundc1", "Kdm6a", "Uba1", "Cdk16", "L1cam", "Arhgap4", "Naa10", "Renbp","Hcfc1","Ikbkg", "Gab3" , "Tbl1x" ,  "Prkx", "Zfx","Eif2s3x", "Tsix","Xist", # "Sh3bgrl", "Chm","Nap1l3" , "Sytl4", "Morf4l2", "Alg13","Ribc1","Smc1a","Kdm5c","Eif1ax" , "Rbbp7" ,  "Syap1", "S100g", "Pir","Gemin8", "Gpm6b","Ofd1", "Rab9", "Tmsb4x", #"Clcn4","Dazl","Kdm5d","Eif2s3y", "Uty","Ddx3y",  "Car5b")
```


```{r}
fl_samples_E14.5_12May_named_subset_regressed <- fl_samples_E14.5_12May_named_subset_nosexdoublets
# Add sex gene scores to metadata
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "RNA"
fl_samples_E14.5_12May_named_subset_regressed <- AddModuleScore(object = fl_samples_E14.5_12May_named_subset_regressed, features = sex_genes, name = "Sex_Genes")
fl_samples_E14.5_12May_named_subset_regressed[]
```


```{r}
# SCTransform the subset Ery object
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "RNA"
fl_samples_E14.5_12May_named_subset_regressed <- SCTransform(fl_samples_E14.5_12May_named_subset_regressed,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score", "Sex_Genes1", "sex"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



## Reannotate clusters in FL_E12.5 samples
```{r}
# Fetal liver
fl_samples_E14.5_12May_named_subset_regressed <-  RenameIdents(object = fl_samples_E14.5_12May_named_subset_regressed,
                                      "EP1b" = "EP2",
                                      "EP2a" = "EP1",
                                      "EP3a" = "EP1",
                                      "EP2b" = "EP2",
                                      "EP1a" = "EP1",
                                      "EP4a" = "EP1",
                                      "EP5a" = "EP1")

fl_samples_E14.5_12May_named_subset_regressed$cluster_ident <- Idents(fl_samples_E14.5_12May_named_subset_regressed)


```


```{r}
fl_samples_E14.5_12May_named_subset_regressed@active.ident <- factor(fl_samples_E14.5_12May_named_subset_regressed@active.ident, levels = c("MEP", "Mk", "EP1", "EP2", "PEBa", "PEBb", "PEBc"))
```


```{r, eval=FALSE}
fl_samples_E14.5_12May_named_subset_regressed$cluster_ident <- factor(fl_samples_E14.5_12May_named_subset_regressed$cluster_ident, levels = c("MEP", "Mk", "EP1", "EP2", "PEBa", "PEBb", "PEBc"))
```





```{r}
# Set colors
clust_cols_subset <- c(MEP = "#565656", Mk = "brown", EP1 = "#DEA0FD", EP2 = "#2ED9FF", PEBa = "#F8766D",  PEBb= "#3283FE", PEBc = "#FE00FA")
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        cols = clust_cols_subset,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend() +
        theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())

ggsave("repeated_umap2_fl_samples_E14.5_12May_named_subset_regressed_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Plot UMAP
set.seed(12098)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        cols = clust_cols_subset,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend()

ggsave("R_umap_fl_samples_E14.5_12May_named_subset_regressed.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```




```{r}
# Run PrepSCTFindMarkers to ensure that the fixed value is set properly, as SCT creates corrected counts by setting the sequencing depth for all the cells to a fixed value
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "integrated"
fl_samples_E14.5_12May_named_subset_regressed <- PrepSCTFindMarkers(fl_samples_E14.5_12May_named_subset_regressed) 
```


```{r, eval=FALSE}
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers_fl_samples_E14.5_12May_named_subset_regressed <- FindAllMarkers(object = fl_samples_E14.5_12May_named_subset_regressed, 
                          only.pos = TRUE,
                          assay = "SCT",
                          logfc.threshold = 0.25,
                          min.pct = 0.25,
                          thresh.use = 0.25)
```


```{r, eval=FALSE} 
# Arrange markers by average log2fold change
markers_fl_samples_E14.5_12May_named_subset_regressed <- markers_fl_samples_E14.5_12May_named_subset_regressed %>% 
  dplyr::group_by(cluster) %>% 
  dplyr::arrange(desc(avg_log2FC), .by_group = TRUE)

```


```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "SCT"
# Let us look at the top markers of each cluster
top_3_fl_samples_E14.5_12May_named_subset_regressed <- markers_fl_samples_E14.5_12May_named_subset_regressed %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 3)

top_5_markers_fl_samples_E14.5_12May_named_subset_regressed <- markers_fl_samples_E14.5_12May_named_subset_regressed %>% group_by(cluster) %>% 
  slice_max(avg_log2FC, n = 5)
```



# Plot top 3 genes
```{r}
top_3_markers <- top_3_fl_samples_E14.5_12May_named_subset_regressed %>% 
                  ungroup() %>%
                  dplyr::select(gene) %>% 
                  unique() %>% 
                  .[!is.na(.)]

mat <- fl_samples_E14.5_12May_named_subset_regressed@assays$SCT@data[top_3_markers, ] %>% as.matrix()
mat <- t(scale(t(mat)))

cols_sample_group <- c(FL_E14.5 = "#E2E2E2" )

clust <- c("MEP", "Mk", "EP1", "EP2", "PEBa", "PEBb", "PEBc")

clusters <- fl_samples_E14.5_12May_named_subset_regressed$cluster_ident

                
annotation_col <-  data.frame(E14.5=clusters)

annotation_colors <- list(E14.5 = clust_cols_subset,
                          Timepoint=cols_sample_group)
                                        
# Determine expression ranges in the matrix
quantile(mat, c(0.1, 0.95))
cols <- circlize::colorRamp2(c(-1, 0, 2), c("blue", "black", "yellow"))


pdf(file="~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots/R_top3_heatmap_fl_samples_E14.5_12May_named_subset_regressed.pdf", width = 25, height = 25)
heat <- ComplexHeatmap::Heatmap(mat, name="E14.5 Top3",
                        column_split = factor(clusters, levels = clust),
                        cluster_columns = TRUE,
                        show_column_dend = FALSE,
                        show_column_names = FALSE,
                        column_title = clust,
                        cluster_column_slices = FALSE,
                        column_title_gp = gpar(fontsize = 20),
                        row_names_gp = gpar(fontsize = 25),
                        column_gap = unit(0.5, "mm"),
                        cluster_rows = TRUE,
                        show_row_dend = FALSE,
                        col = cols,
                        row_gap = unit(10, "mm"),
                        column_title_rot = 60,
                        top_annotation = HeatmapAnnotation(df = annotation_col, col = annotation_colors, show_legend = FALSE, show_annotation_name = FALSE),
                        heatmap_legend_param = list(labels_gp = gpar(fontsize = 30), title_gp = gpar(fontsize = 30, fontface = "bold")),
                        width = 25,
                        height = 25,
                        use_raster = TRUE,
                        raster_quality = 4)
draw(heat)
dev.off()
#annotation_legend_param = list(labels_gp = gpar(fontsize = 30),show_annotation_name = TRUE,title_gp = gpar(fontsize = 30, fontface = "bold")

```




```{r}
#library(viridis)
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "RNA"
fl_samples_E14.5_12May_named_subset_regressed <- ScaleData(fl_samples_E14.5_12May_named_subset_regressed)
```


```{r}
mapal <- colorRampPalette(RColorBrewer::brewer.pal(11,"RdBu"))(256)
DoHeatmap(subset(fl_samples_E14.5_12May_named_subset_regressed, downsample = 100),
          features = top_3_markers,
          group.by = "cluster_ident",
          group.colors = c(MEP = "#565656", Mk = "brown", EP1 = "#DEA0FD", EP2 = "#2ED9FF", PEBa = "#F8766D",  PEBb= "#3283FE", PEBc = "#FE00FA"), 
          angle = 45) +
          theme(text = element_text(size = 30)) +
  scale_fill_gradientn(colours = rev(mapal))
          #scale_fill_viridis()
          
  ggsave("Top3_heatmap_fl_samples_E14.5_12May_named_subset_regressed_named.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```

```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        cols = clust_cols_subset,
        label = TRUE,
        label.size = 11,
        pt.size = 1.5) +
        #theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.ticks = element_blank()) +
        theme(axis.line = element_blank()) 
        #panel.background = element_rect(fill = "white") +
        #theme(plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) 

ggsave("repeated_labelled_umap2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Plot UMAP
set.seed(123)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        cols = clust_cols_subset,
        label = FALSE,
        label.size = 11,
        pt.size = 1.5) + NoLegend() +
        #theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.ticks = element_blank()) +
        theme(axis.line = element_blank()) 
        #panel.background = element_rect(fill = "white") +
        #theme(plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) 

ggsave("repeated_umap2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
(ncells_E14.5 <- data.frame(table(fl_samples_E14.5_12May_named_subset_regressed$cluster_ident)) %>% dplyr::rename("Clusters" = Var1))

(ncells_E14.5 <- mutate(ncells_E14.5, percent = round((Freq/sum(Freq))*100, digits = 2)))
ncells_E14.5$percent <- paste0(ncells_E14.5$percent,"%")
ncells_E14.5

ggplot(ncells_E14.5, aes(x = Clusters, y = Freq, fill = Clusters)) +
  geom_col() +
  scale_fill_manual(values = c(MEP = "#565656", Mk = "brown", EP1 = "#DEA0FD", EP2 = "#2ED9FF", PEBa = "#F8766D",  PEBb= "#3283FE", PEBc = "#FE00FA")) +
  theme(axis.text = element_text(size = 50),
        panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) +
        theme(axis.title.x = element_text(size =50, face = "bold")) +
        theme(axis.title.y = element_text(size = 50, face = "bold")) +
  geom_text(aes(label = percent), vjust = 0.5, hjust = -0.1, size = 15, face = "bold") +
  #scale_y_continuous(breaks = seq(0, 2000, by=1000), limits=c(0,2000)) +
  scale_y_continuous(breaks = seq(0, 4000, by=1000), limits=c(0,4000)) +
  xlab("Cell types") + 
 # theme_bw() + 
  NoLegend() +
  coord_flip()

ggsave("repeated_cellsincluster_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```

```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "RNA"

# First, subset object into replicates
E14.5_rep1 <- subset(x = fl_samples_E14.5_12May_named_subset_regressed, subset = sample == "FL_E14.5_rep_1")
E14.5_rep2 <- subset(x = fl_samples_E14.5_12May_named_subset_regressed, subset = sample == "FL_E14.5_rep_2")

# Average genes per cell in each object
(ave_genes_percell_e14.5_rep1 <- mean(Matrix::colSums(GetAssayData(E14.5_rep1, slot = 'counts') > 0)))
(ave_genes_percell_e14.5_rep2 <- mean(Matrix::colSums(GetAssayData(E14.5_rep2, slot = 'counts') > 0)))

# Get average reads/counts per cell
(ave_counts_percell_e14.5_rep1 <- mean(E14.5_rep1@meta.data$nUMI))
(ave_counts_percell_e14.5_rep2 <- mean(E14.5_rep2@meta.data$nUMI))

rm(E14.5_rep1, E14.5_rep2)
```

[1] 4082.461
[1] 4089.602
[1] 17372.24
[1] 17619.93



```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "SCT"

# Megakaryocyte progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("F2r"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Itga2b"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Gp1bb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
 
a
ggsave("subset_Mk_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_Mk_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_Mk_marker3_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
# Megakaryocytes
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Myl9"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Vwf"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Ppbp"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Alox12"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Zyx"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("subset_Mk_marker4_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_Mk_marker5_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_Mk_marker6_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("subset_Mk_marker8_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("subset_Mk_marker7_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Gata2"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Klf1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("subset_MEP_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_MEP_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_MEP_marker3_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Megakaryocyte-Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Epor"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Car1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Muc13"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
 
a
ggsave("subset_MEP_marker4_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_MEP_marker5_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_MEP_marker6_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```




```{r}
# Erythroid progenitors
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Kit"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Gata1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Myb"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("subset_EryP_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_EryP_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_EryP_marker3_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("subset_EryP_marker4_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
# Erythroblasts
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Hbb-bs"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Hbb-bt"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

c <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Gypa"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

d <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Hba-a1"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

e <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Tfrc"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

f <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Cd24a"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

a
ggsave("subset_Erythb_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_Erythb_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
c
ggsave("subset_Erythb_marker3_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
d
ggsave("subset_Erythb_marker4_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
e
ggsave("subset_Erythb_marker5_fl_samples_E14.5_12May_named_subset_regressed.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
f
ggsave("subset_Erythb_marker6_fl_samples_E14.5_12May_named_subset_regressed.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```




# LOOK AT SOME Y AND X CHROMOSOME GENES

```{r}
# Male genes
a <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Eif2s3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

b <- FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Ddx3y"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())
a
ggsave("subset_Male_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
b
ggsave("subset_Male_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```



```{r}
DefaultAssay(fl_samples_E14.5_12May_named_subset_regressed) <- "SCT"
# Female gene
FeaturePlot(fl_samples_E14.5_12May_named_subset_regressed, 
            reduction = "umap", 
            features = c("Xist"),
            order = TRUE,
            min.cutoff = 'q10', 
            cols = c("grey", "red"),
            label = TRUE,
            label.size = 11,
            pt.size = 1.5) +
            theme(plot.title = element_text(size = 50, face = "bold")) +
            theme(axis.text.x = element_blank()) +
            theme(axis.text.y = element_blank()) +
            theme(axis.title.x = element_blank()) +
            theme(axis.title.y = element_blank()) +
            theme(axis.line = element_blank()) +
            theme(axis.ticks = element_blank())

ggsave("subset_Female_marker1_fl_samples_E14.5_12May_named_subset_regressed.tif",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15, device = "tiff")
```


```{r}
library(scCustomize)

#tiff("~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots/Clustered_Dotplot_marker2_fl_samples_E14.5_12May_named_subset_regressed.tif", width = 550, height = 550)
Clustered_DotPlot(seurat_object = fl_samples_E14.5_12May_named_subset_regressed, features = c("Gypa", "Tfrc", "Gata1", "Kit", "Epor", "Gata2", "Pf4", "Vwf", "Itga2b", "Thbs1", "Gata2", "Gata1", "Klf1", "Car1", "Ppbp", "Hbb-bs", "Hbb-bt", "Hba-a2", "Rhd", "Gp1bb", "Cd24a", "Muc13", "Myb", "Epor", "F2r", "Myl9", "Pf4", "Alox12"), k = 3, show_parent_dend_line = FALSE, plot_km_elbow = FALSE, cluster_ident = FALSE,  column_label_size = 10, row_label_size = 20, colors_use_idents = c(MEP = "#565656", Mk = "brown", EP1 = "#DEA0FD", EP2 = "#2ED9FF", PEBa = "#F8766D",  PEBb= "#3283FE", PEBc = "#FE00FA"), colors_use_exp = c(low="lightgrey", mid="grey", high="red")) 

#row_label_fontface = "bold",
dev.off()
```


```{r}
# Plot UMAP
set.seed(1203)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        cols = clust_cols_subset,
        split.by = "sample",
        label.size = 6,
        pt.size = 1.5) +
        theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())
ggsave("subset_replicates_color_umap_fl_samples_E14.5_12May_named_subset_regressed.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```


```{r}
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed,
        reduction = "umap",
        group.by = "sex",
        label.size = 6,
        pt.size = 1.5) +
        theme(plot.title = element_text(size = 50, face = "bold")) +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())
```


I WILL DO TRAJECTORY INFERENCE USING CELLRANK AND NOT PALANTIR (CELLRANK WORKS BETTER FOR THIS DATA) BEFORE I COME BACK TO DIFFERENTIAL GENE EXPRESSION ANALYSIS AND PATHWAY ANALYSIS

```{r}
#dir.create("~/myproject/sc_analysis_project/msc_project/rnavelocity/E14.5_rnavelocity")
E14.5_rnavelo <- "~/myproject/sc_analysis_project/msc_project/rnavelocity/E14.5_rnavelocity/"
```


## PREPARE ERYTHROID SEURAT OBJECT METADATA FOR RNA VELOCITY (https://smorabit.github.io/tutorials/8_velocyto/)

## Ery
```{r}
# Make a copy of the original object to avoid stories that touch
fl_samples_E14.5_12May_named_subset_regressed1 <- fl_samples_E14.5_12May_named_subset_regressed
# save metadata table
fl_samples_E14.5_12May_named_subset_regressed1$barcode <- colnames(fl_samples_E14.5_12May_named_subset_regressed1)
fl_samples_E14.5_12May_named_subset_regressed1$UMAP_1 <- fl_samples_E14.5_12May_named_subset_regressed1@reductions$umap@cell.embeddings[,1]
fl_samples_E14.5_12May_named_subset_regressed1$UMAP_2 <- fl_samples_E14.5_12May_named_subset_regressed1@reductions$umap@cell.embeddings[,2]
```  


```{r}


# Select column variables of interest
Ery_E14.5_metadata <- fl_samples_E14.5_12May_named_subset_regressed1@meta.data
Ery_E14.5_metadata <- Ery_E14.5_metadata %>% dplyr::select(c(seq_folder, cells, sample, sample_group, sample_origin, S.Score, G2M.Score, Phase, cluster_ident, barcode, UMAP_1, UMAP_2)) %>% dplyr::rename("CellID" = cells)

fl_samples_E14.5_12May_named_subset_regressed1@meta.data <- Ery_E14.5_metadata
```

```{r}
write.csv(fl_samples_E14.5_12May_named_subset_regressed1@meta.data, file = paste0(E14.5_rnavelo, "Ery_E14.5_metadata.csv"), quote = F, row.names = F)

```


```{r}
# Write expression counts matrix
Ery_E14.5_counts_matrix <- GetAssayData(fl_samples_E14.5_12May_named_subset_regressed1, assay = "SCT", slot = "counts")
writeMM(Ery_E14.5_counts_matrix, file = paste0(E14.5_rnavelo, "Ery_E14.5_counts.mtx"))
```

```{r}
# Write dimensionality reduction matrix
write.csv(fl_samples_E14.5_12May_named_subset_regressed1@reductions$pca@cell.embeddings, file = paste0(E14.5_rnavelo, "Ery_E14.5_pca.csv"), quote = F, row.names = F)
```


```{r}
# Write gene names
write.table(data.frame("gene" = rownames(Ery_E14.5_counts_matrix)), file = paste0(E14.5_rnavelo, "Ery_E14.5_gene_names.csv"), quote = F, row.names = F, col.names = F)
```



# NOTE:

# THE FOLLOWING ANNOTATION WAS DONE BEFORE I REPEATED ANNOTATION ABOVE AND GROUPED SUBCLUSTERS OF ERYTHROID PROGENITORS


## Subset Erythroid progenitors

## Subset clusters of interest. 

```{r, eval=FALSE}
fl_samples_E14.5_12May_named_subset_regressed_named <- fl_samples_E14.5_12May_named_subset_regressed
fl_samples_E14.5_12May_named_subset_regressed_named_EPs <- subset(fl_samples_E14.5_12May_named_subset_regressed_named, idents =c("EP1a", "EP2a", "EP3a", "EP4a", "EP5a", "EP1b", "EP2b"))
 
```



```{r}
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset_regressed_named_EPs <- SCTransform(fl_samples_E14.5_12May_named_subset_regressed_named_EPs,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score", "Sex_Genes1", "sex"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```

# NOTE:

# THE FOLLOWING ANNOTATION WAS DONE BEFORE I REPEATED ANNOTATION ABOVE AND GROUPED SUBCLUSTERS OF ERYTHROID PROGENITORS
## Annotate clusters in FL_E12.5 samples for DEG analysis between erythroid progenitors

```{r}
# Fetal liver
fl_samples_E14.5_12May_named_subset_regressed_named_EPs <-  RenameIdents(object = fl_samples_E14.5_12May_named_subset_regressed_named_EPs,
                                                  "EP1a" = "EPa",
                                                  "EP1b" = "EPb",
                                                  "EP2a" = "EPa",
                                                  "EP2b" = "EPb",
                                                  "EP3a" = "EPa",
                                                  "EP4a" = "EPa",
                                                  "EP5a" = "EPa")

fl_samples_E14.5_12May_named_subset_regressed_named_EPs$cluster_ident <- Idents(fl_samples_E14.5_12May_named_subset_regressed_named_EPs)

```


```{r}
table(fl_samples_E14.5_12May_named_subset_regressed_named_EPs$cluster_ident, fl_samples_E14.5_12May_named_subset_regressed_named_EPs$sample_group)
```
     FL_E14.5
  EPa     3061
  EPb     1969


```{r}
# Plot UMAP
set.seed(12098)
DimPlot(object = fl_samples_E14.5_12May_named_subset_regressed_named_EPs,
        reduction = "umap",
        cols = clust_cols_EP,
        label = TRUE,
        label.size = 6,
        pt.size = 1.5) + NoLegend()

ggsave("R_umap_EP_fl_samples_E14.5_12May_named_subset_regressed_named_EPs.pdf",
       path = "~/myproject/sc_analysis_project/msc_project/results/plots/FL_E14.5/12May_plots",
       width = 17, height = 15)
```



# DIFFERENTIAL GENE EXPRESSION BETWEEN EPa and EPb. EPa represents Females, EPb represents males

```{r}
female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets <- FindMarkers(fl_samples_E14.5_12May_named_subset_regressed_named_EPs, 
                                  ident.1 = "EPa",
                                  ident.2 = "EPb",
                                   logfc.threshold = 0,
                                  min.cells.feature = 3,
                                  min.cells.group = 3,
                                  min.pct = 0.1)

female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets
```


```{r}
# Arrange adjusted p-values
female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets %>% 
  dplyr::arrange("p_val_adj") %>% 
  dplyr::mutate(gene = rownames(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets))

# What is the total number of genes in the DE analysis?
nrow(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets)

# How many genes have significant adjusted p-values,  that is differentially expressed?
female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets %>% 
                              dplyr::filter(p_val_adj < 0.05)
nrow(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif)

# How many genes are upregulated?
(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif %>% 
                              dplyr::filter(avg_log2FC > 0))
nrow(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg)

# How many genes are down-regulated?
(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif %>% 
                              dplyr::filter(avg_log2FC < 0))
nrow(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg)
```



```{r}
# Save the DEGs (all genes, not just the significant ones)
saveRDS(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets.rds")))
write.csv(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets.csv")))

# Save significant DEGs
saveRDS(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.rds")))
write.csv(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.csv")))

# Save upregulated genes
saveRDS(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg.rds")))
write.csv(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_upreg.csv")))

# Save downregulated genes
saveRDS(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg.rds")))
write.csv(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg,
          paste0(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_downreg.csv")))
```



```{r}
EnhancedVolcano(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif, 
                x = "avg_log2FC", 
                y = "p_val_adj", 
                xlim = c(min(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif[["avg_log2FC"]], na.rm = TRUE) - 0.5, max(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif[["avg_log2FC"]], na.rm = TRUE) + 0.5),
                lab = rownames(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif),
                pCutoff = 10e-5,
                pointSize = 6,
                labSize = 10,
                drawConnectors = TRUE,
                #max.overlaps = Inf,
                labFace = "bold",
                axisLabSize = 30,
                subtitleLabSize = 30,
                legendLabSize = 30,
                legendIconSize = 15,
                captionLabSize = 30,
                FCcutoff = 0,
                col = c("grey30", "forestgreen", "dodgerblue4", "red4"),
                title = "", 
                ylab = "-log10(p-val adj)", 
                subtitle = "E14.5 female vs male erythroid progenitor gene expression", 
                legendLabels = c("Not sig.", expression(Log[2] ~ FC), "p-value", expression(p - value ~ +
                ~ log[2] ~ FC)))

#ggsave("volcano_female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets.pdf",
      # path = "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/",
       #width = 15, height = 15)
```

```{r}
# Read in up- and down-regulated Erythroid progenitor genes
female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif <- readRDS(here("DE_analysis/results/FL_E14.5", "female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.rds"))

#female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif <- read_csv(here("DE_analysis/results/FL_E14.5", #"female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.csv"))
```


# Add annotations (entrezid and ensembl IDs to genes) for GO analysis for EP genes

```{r}
ah <- AnnotationHub()
```


```{r}
unique(ah$species) %>% View()
unique(ah$rdataclass) %>% View()
unique(ah$dataprovider)
```


```{r}
# Access the Ensembl database for organism of interest
AhDb <- query(ah, 
              pattern = c("Mus musculus", "EnsDb"),
              ignore.case = TRUE)
```


```{r}
# Extract annotations of interest
mouse_ens <- AhDb[["AH60788"]]

#d <- display(ah) # open web browser for ah
```


```{r}
# Extract gene-level information from database
annotations <- genes(mouse_ens, 
                     return.type = "data.frame")

# Select annotations of interest
annotations <- annotations %>% 
  dplyr::select(gene_id, gene_name, entrezid, symbol, gene_biotype)

# Filter dataframe to just our gene results table. Don't do this so you can use annotations for other celltype DEGs
#annotations <- annotations %>% 
  #dplyr::filter(gene_name %in% EPa_vs_EPb_E12.5_regressedsexandsexgenes_nosexdoublets_upreg)

# Check class of annotations$entrezid
class(annotations$entrezid)

# Check if multiple genes have the same entrezid
which(map(annotations$entrezid, length) > 1) # More than one match have the same percentage, so they are returned together

# Keep the first identifier - entrezid
annotations$entrezid <- map(annotations$entrezid, 1) %>% unlist()
```


```{r}
# How many Ensembl identifiers have an associated gene symbol
which(is.na(annotations$gene_name)) %>% length()

# How many of them are unique
which(duplicated(annotations$gene_name)) %>% length()

# Determine the indices for the non-duplicated genes
non_duplicated_idx <- which(duplicated(annotations$gene_name) == FALSE)

# How many rows does annotations have?
annotations %>% nrow()

# Return only the non-duplicated genes using the indices
annotations <- annotations[non_duplicated_idx, ]

# How many rows does annotations now have?
annotations %>% nrow()

# Determine how many of the Entrez column entries are NA
which(is.na(annotations$entrezid)) %>% length()

# Save annotations object
#write_csv(annotations,
          #paste0(here("DE_analysis/results/", "gene_annotations.csv")))
```


```{r}
# Merge annotations dataframe with our significant genes
female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif_anno <- left_join(female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif, annotations, by = c("gene" = "gene_name"))
```



```{r}
# Will use all significant genes
# Get the log2 fold change
gene_list <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif_anno$avg_log2FC

# Name the vector
names(gene_list) <- female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif_anno$gene

# Omit NA values
#gene_list <- na.omit(names(gene_list))

# Sort the list in decreasing order (clusterProfiler requirement)
gene_list <- sort(gene_list, decreasing = TRUE)
```



```{r}
# Create a data frame using relevant columns
mydf <- data.frame(gene = names(gene_list), FC = gene_list)
mydf$group <- with(mydf, ifelse(FC > 0, "Female", "Male"))
```



```{r}
enrichedterm_res <- compareCluster(gene~group, data = mydf, fun = "enrichGO", keyType = 'SYMBOL', OrgDb = org.Mm.eg.db, ont = "BP", pAdjustMethod = "fdr", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
```


```{r}
as.data.frame(enrichedterm_res)
```


```{r}
categories1 <- c("cellular respiration", "mitochondrial respiratory chain complex assembly", "protein folding", "response to endoplasmic reticulum stress", "translational elongation","electron transport chain",  "ribosome biogenesis", "cellular aldehyde metabolic process", "regulation of cellular response to oxidative stress", "cellular response to toxic substance")
clusterProfiler::dotplot(enrichedterm_res, showCategory = categories1, size = "count") +
  theme(axis.text.x = element_text(face = "bold", size = 25)) +
theme(axis.text.y = element_text(face = "bold", size = 30)) +
theme(plot.title = element_text(size = 25, face = "bold")) +
theme(axis.title.x = element_text(size = 25, face = "bold")) +
theme(legend.text = element_text(size = 25, face = "bold")) +
theme(legend.title = element_text(size = 25, face = "bold")) +
guides(color = guide_legend(override.aes = list(size = 5)))

ggsave("GO_female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.tif",
       path = "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/",
       width = 15, height = 15, device = "tiff")
```







```{r}
# Extract databases of interest
msigdbr_df1 <- msigdbr(species = "mouse", category = "C2", subcategory = "CP:REACTOME")
msigdbr_df2 <- msigdbr(species = "mouse", category = "C5", subcategory = "GO:BP")


# Extract columns of interest(gene set name and gene symbol)
# Reactome Pathways
msigdbr_t2g1 <- msigdbr_df1 %>% dplyr::distinct(gs_name, gene_symbol) %>% as.data.frame()
msigdbr_t2g1$gs_name <- msigdbr_t2g1$gs_name %>% str_replace_all("REACTOME_", "")
# Biological Processes
msigdbr_t2g2 <- msigdbr_df2 %>% dplyr::distinct(gs_name, gene_symbol) %>% as.data.frame()
```



# USE REACTOME
```{r}
enrichedreactome_res <- compareCluster(gene~group, data = mydf, fun = "enricher", TERM2GENE = msigdbr_t2g1,  pAdjustMethod = "fdr", pvalueCutoff = .1, qvalueCutoff = .1)
```



```{r}
as.data.frame(enrichedreactome_res)
```


```{r}
categories2 <- c("CELLULAR_RESPONSE_TO_CHEMICAL_STRESS", "RESPIRATORY_ELECTRON_TRANSPORT", "CELLULAR_RESPONSE_TO_HYPOXIA", "SYNTHESIS_OF_DNA", 
                 "TRANSLATION", "COMPLEX_I_BIOGENESIS", "THE_CITRIC_ACID_TCA_CYCLE_AND_RESPIRATORY_ELECTRON_TRANSPORT","PROTEIN_LOCALIZATION", "MITOCHONDRIAL_PROTEIN_IMPORT","PROTEIN_LOCALIZATION","CYTOPROTECTION_BY_HMOX1")

clusterProfiler::dotplot(enrichedreactome_res, showCategory = categories2, size = "count") +
  theme(axis.text.x = element_text(face = "bold", size = 25)) +
theme(axis.text.y = element_text(face = "bold", size = 25)) +
theme(plot.title = element_text(size = 25, face = "bold")) +
theme(axis.title.x = element_text(size = 25, face = "bold")) +
theme(legend.text = element_text(size = 25, face = "bold")) +
theme(legend.title = element_text(size = 25, face = "bold")) +
guides(color = guide_legend(override.aes = list(size = 5)))


ggsave("REACTOME_female_vs_male_E14.5_regressedsexandsexgenes_nosexdoublets_signif.tif",
       path = "~/myproject/sc_analysis_project/msc_project/DE_analysis/figures/FL_E14.5/",
       width = 15, height = 15, device = "tiff")
```




# Filter metadata to remove cells with unknown sex
```{r}
fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known <- subset(fl_samples_E14.5_12May_named_subset_regressed_named_EPs, subset = sex != "unknown")
```




```{r}
# SCTransform the subset Ery object
fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known <- SCTransform(fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known,
                                   vst.flavor = "v2",
                                   vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score", "Sex_Genes1", "sex"),
                                   variable.features.n = 3000,
                                   verbose = FALSE)
```



```{r}
# Table of cells based on sex
table(fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known$sex, fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known$cluster_ident)

table(fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known$sex, fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known$cluster_ident)
```



# TRANSCRIPTION FACTOR ENRICHMENT ANALYSIS USING SCENIC

# Run the analysis on Compute Canada
# Use the Rscript ~/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/computecanadascenicE14.5.R


```{r}
# Create a directory for transferring Compute Canada output since you do not want to transfer input files
#dir.create("/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs")
```



```{r}
# Read in already saved scenicOptions object (computed on Compute Canada)
scenicOptions_E14.5_EPs <- readRDS("/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/int/scenicOptions_E14.5_EPs.rds")
fl_samples_E14.5_sex_scenic_cellInfo <- readRDS("/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/int/fl_samples_E14.5_sex_scenic_cellInfo.rds")
scenicOptions_E14.5_EPs@inputDatasetInfo$cellInfo <- "/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/int/fl_samples_E14.5_sex_scenic_cellInfo.rds"
scenicOptions_E14.5_EPs@settings$dbDir <- "/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs"
```


```{r}
# Save cellInfo
saveRDS(fl_samples_E14.5_sex_scenic_cellInfo, "/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/int/fl_samples_E14.5_sex_scenic_cellInfo.rds")
```



```{r}
# Add cellInfo to scenicOptions object
scenicOptions_E14.5_EPs@inputDatasetInfo$cellInfo <- "/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/int/fl_samples_E14.5_sex_scenic_cellInfo.rds"
```



```{r}
setwd("/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs")
set.seed(181)
regulons <- loadInt(scenicOptions_E14.5_EPs, "regulons")

regulonAUC <- loadInt(scenicOptions_E14.5_EPs, "aucell_regulonAUC")

# Extract AUC scores of regulons 
regulon_auc_score_E14.5_EPs <- getAUC(regulonAUC)

regulonAUC <- regulonAUC[onlyNonDuplicatedExtended(rownames(regulonAUC)),]
regulonActivity_bySex <- sapply(split(rownames(fl_samples_E14.5_sex_scenic_cellInfo), fl_samples_E14.5_sex_scenic_cellInfo$sex),
                                     function(allcells) rowMeans(getAUC(regulonAUC)[,allcells]))
regulonActivity_bySex_Scaled <- t(scale(t(regulonActivity_bySex), center = T, scale=T))

regulonActivity_bySex_Scaled <- regulonActivity_bySex_Scaled %>% as.data.frame() %>% dplyr::select(-unknown) %>% as.matrix()


cols_sex <- c(Male = "#7997FF", Female = "pink2")


annotation_col <-  data.frame(Sex=unique(fl_samples_E14.5_sex_scenic_cellInfo$sex),
                              Timepoint=unique(fl_samples_E14.5_sex_scenic_cellInfo$sample_group))
                              #Sex=fl_samples_E14.5_sex_scenic_cellInfo$sex)
annotation_colors <- list(Sex=cols_sex)

rownames(regulonActivity_bySex_Scaled) <- gsub("_extended", "", rownames(regulonActivity_bySex_Scaled))
                                        

#pdf(file="/Users/ekperekaamutaigwe/myproject/sc_analysis_project/msc_project/cisTarget_databases_E14.5_EPs/fl_samples_E14.5_sex_scenic_heatmap.pdf", width = 25, height =25)
heat1 <- ComplexHeatmap::Heatmap(regulonActivity_bySex_Scaled, name="Regulon act.",
                        cluster_columns = FALSE,
                        show_column_dend = FALSE,
                        cluster_column_slices = FALSE,
                        #column_title_gp = gpar(fontsize = 8),
                        #column_gap = unit(0.5, "mm"),
                        row_names_gp = gpar(fontsize = 6),
                        cluster_rows = TRUE,
                        show_row_dend = FALSE,
                        #col = viridisLite::viridis(length(mat_breaks) - 1),
                        #row_names_gp = gpar(fonsize = 4),
                        column_title_rot = 45,
                        top_annotation = HeatmapAnnotation(df = annotation_col, col = annotation_colors, show_legend = TRUE, show_annotation_name = FALSE, annotation_legend_param = list(labels_gp = gpar(fontsize = 6),  title_gp = gpar(fontsize = 6, fontface = "bold"))),
                        show_column_names = FALSE,
                         heatmap_legend_param = list(labels_gp = gpar(fontsize = 6), title_gp = gpar(fontsize = 6, fontface = "bold")),
                        width = 25,
                        height = 25,
                        use_raster = TRUE,
                        raster_quality = 4)
draw(heat1)
#dev.off()

```


```{r, eval=FALSE}
# Save integrated clustered and identified FL_E14.5 data

# Sept 10th
saveRDS(fl_samples_E14.5_12May, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_clustered_sexgenesnotremoved.rds")
saveRDS(markers_fl_samples_E14.5_12May_named, "~/myproject/sc_analysis_project/msc_project/results/FL_E14.5/markers_fl_samples_E14.5_12May_named.rds")
write_csv(markers_fl_samples_E14.5_12May_named, "~/myproject/sc_analysis_project/msc_project/results/FL_E14.5/markers_fl_samples_E14.5_12May_named.csv", quote = "none")

saveRDS(fl_samples_E14.5_12May_named_subset, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/repeated_fl_samples_E14.5_12May_named_subset.rds")
saveRDS(fl_samples_E14.5_12May_named_subset_nosexdoublets, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_nosexdoublets_notregressed.rds")
saveRDS(fl_samples_E14.5_12May_named_subset_regressed, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressedsexandsexgenes.rds")

saveRDS(fl_samples_E14.5_12May_named_subset_regressed, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressed_MERGED_EPs.rds") # FEB 25, 2024

saveRDS(markers_fl_samples_E14.5_12May_named_subset_regressed, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/markers_fl_samples_E14.5_12May_named_subset_regressed_MERGED_EPs.rds")

saveRDS(fl_samples_E14.5_12May_named_subset_regressed_named, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressedsexandsexgenes_named.rds")

saveRDS(fl_samples_E14.5_12May_named_subset_regressed_named_EPs, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressed_named_EPs.rds")
saveRDS(fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_regressed_named_EPs_sex_known.rds")


#saveRDS(markers_fl_samples_E12.5_12May_named_subset, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E12.5/markers_fl_samples_E12.5_12May_named_subset.rds")
saveRDS(fl_samples_E14.5_12May_named_subset_DEG, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_DEG.rds")
saveRDS(fl_samples_E14.5_12May_named_subset_DEG_merge, "~/myproject/sc_analysis_project/msc_project/rds_data/FL_E14.5/fl_samples_E14.5_12May_named_subset_DEG_merge.rds")

```



# MAKE BARPLOTS OF CELL CYCLE PHASE PROPORTIONS ACROSS E10.5, E12.5 AND E14.5

## Merge objects from different time points. CLUSTERED CELLS 
```{r}
# Merge fetal liver samples for downstream analysis. 
merged_timepoints <- merge(x = fl_samples_E10.5_12May, y = list(fl_samples_E12.5_12May, fl_samples_E14.5_12May), merge.dr = "TRUE")
```

```{r}
VariableFeatures(merged_timepoints[["SCT"]]) <- rownames(merged_timepoints[["SCT"]]@scale.data)
```


```{r}
library(ggplot2)
metadata <- merged_timepoints@meta.data
  ggplot(metadata, aes(x = sample_group, fill = Phase)) +
        geom_bar() +
    scale_y_continuous(breaks = seq(0, 10000, by=2000), limits=c(0,10000)) +
    theme(axis.text = element_text(size = 50),
            panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 0)) +
    theme(legend.text = element_text(size = 30)) +
    theme(legend.title = element_text(size = 30)) +
        coord_flip() + 
        theme(axis.title.y = element_blank()) +
        theme(axis.title.x = element_text(size = 30)) 
         #xlab("") 


ggsave("proportion_of_cells_byphase_acrosstimepoints.tif", 
       path = "~/myproject/sc_analysis_project/msc_project/results/plots",
       width = 17, height = 15, device = "tiff")


```




```{r}
library(dittoSeq)
dittoBarPlot(merged_timepoints, "Phase", group.by = "sample_group", colors = c("red", "green4", "#06A5FF")) +
 theme(axis.text = element_text(size = 50),
        panel.background = element_rect(fill = "white"), 
        plot.background = element_rect(fill = "white", colour = "black", linewidth = 1.2)) +
        theme(axis.title.x = element_text(size =50, face = "bold")) +
        theme(axis.title.y = element_text(size = 50, face = "bold")) 
```

```{r}
library(ggplot2)
ggplot(merged_timepoints@meta.data, aes(x = Condition, y = percent, fill = Cell_Type))+
  geom_bar(stat = "identity")+
  geom_text(aes(label = paste(percent,"%")), position = position_stack(vjust = 0.5))
```




```{r}
DefaultAssay(merged_timepoints) <- "SCT"
DimPlot(merged_timepoints,
        reduction = "umap",
        split.by = "sample_group",
        group.by= "Phase") +
        theme(axis.text.x = element_blank()) +
        theme(axis.text.y = element_blank()) +
        theme(axis.title.x = element_blank()) +
        theme(axis.title.y = element_blank()) +
        theme(axis.line = element_blank()) +
        theme(axis.ticks = element_blank())
```






```{r}
sessionInfo()
```


